// fathom-typescript/src/sdk/models/shared/actionitem.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  Assignee,
  Assignee$inboundSchema,
  Assignee$Outbound,
  Assignee$outboundSchema,
} from "./assignee.js";

export type ActionItem = {
  /**
   * Always displayed in English.
   */
  description: string;
  userGenerated: boolean;
  completed: boolean;
  /**
   * Timestamp relative to the start of the recording (HH:MM:SS).
   */
  recordingTimestamp: string;
  recordingPlaybackUrl: string;
  assignee: Assignee;
};

/** @internal */
export const ActionItem$inboundSchema: z.ZodType<
  ActionItem,
  z.ZodTypeDef,
  unknown
> = z.object({
  description: z.string(),
  user_generated: z.boolean(),
  completed: z.boolean(),
  recording_timestamp: z.string(),
  recording_playback_url: z.string(),
  assignee: Assignee$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "user_generated": "userGenerated",
    "recording_timestamp": "recordingTimestamp",
    "recording_playback_url": "recordingPlaybackUrl",
  });
});

/** @internal */
export type ActionItem$Outbound = {
  description: string;
  user_generated: boolean;
  completed: boolean;
  recording_timestamp: string;
  recording_playback_url: string;
  assignee: Assignee$Outbound;
};

/** @internal */
export const ActionItem$outboundSchema: z.ZodType<
  ActionItem$Outbound,
  z.ZodTypeDef,
  ActionItem
> = z.object({
  description: z.string(),
  userGenerated: z.boolean(),
  completed: z.boolean(),
  recordingTimestamp: z.string(),
  recordingPlaybackUrl: z.string(),
  assignee: Assignee$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    userGenerated: "user_generated",
    recordingTimestamp: "recording_timestamp",
    recordingPlaybackUrl: "recording_playback_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ActionItem$ {
  /** @deprecated use `ActionItem$inboundSchema` instead. */
  export const inboundSchema = ActionItem$inboundSchema;
  /** @deprecated use `ActionItem$outboundSchema` instead. */
  export const outboundSchema = ActionItem$outboundSchema;
  /** @deprecated use `ActionItem$Outbound` instead. */
  export type Outbound = ActionItem$Outbound;
}

export function actionItemToJSON(actionItem: ActionItem): string {
  return JSON.stringify(ActionItem$outboundSchema.parse(actionItem));
}

export function actionItemFromJSON(
  jsonString: string,
): SafeParseResult<ActionItem, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ActionItem$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ActionItem' from JSON`,
  );
}
