// fathom-typescript/src/sdk/models/shared/callbackresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type CallbackResponse = {
  destinationUrl: string;
};

/** @internal */
export const CallbackResponse$inboundSchema: z.ZodType<
  CallbackResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  destination_url: z.string(),
}).transform((v) => {
  return remap$(v, {
    "destination_url": "destinationUrl",
  });
});

/** @internal */
export type CallbackResponse$Outbound = {
  destination_url: string;
};

/** @internal */
export const CallbackResponse$outboundSchema: z.ZodType<
  CallbackResponse$Outbound,
  z.ZodTypeDef,
  CallbackResponse
> = z.object({
  destinationUrl: z.string(),
}).transform((v) => {
  return remap$(v, {
    destinationUrl: "destination_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CallbackResponse$ {
  /** @deprecated use `CallbackResponse$inboundSchema` instead. */
  export const inboundSchema = CallbackResponse$inboundSchema;
  /** @deprecated use `CallbackResponse$outboundSchema` instead. */
  export const outboundSchema = CallbackResponse$outboundSchema;
  /** @deprecated use `CallbackResponse$Outbound` instead. */
  export type Outbound = CallbackResponse$Outbound;
}

export function callbackResponseToJSON(
  callbackResponse: CallbackResponse,
): string {
  return JSON.stringify(
    CallbackResponse$outboundSchema.parse(callbackResponse),
  );
}

export function callbackResponseFromJSON(
  jsonString: string,
): SafeParseResult<CallbackResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CallbackResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CallbackResponse' from JSON`,
  );
}
