// fathom-typescript/src/sdk/models/shared/crmcompanymatch.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type CRMCompanyMatch = {
  name: string;
  recordUrl: string;
};

/** @internal */
export const CRMCompanyMatch$inboundSchema: z.ZodType<
  CRMCompanyMatch,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  record_url: z.string(),
}).transform((v) => {
  return remap$(v, {
    "record_url": "recordUrl",
  });
});

/** @internal */
export type CRMCompanyMatch$Outbound = {
  name: string;
  record_url: string;
};

/** @internal */
export const CRMCompanyMatch$outboundSchema: z.ZodType<
  CRMCompanyMatch$Outbound,
  z.ZodTypeDef,
  CRMCompanyMatch
> = z.object({
  name: z.string(),
  recordUrl: z.string(),
}).transform((v) => {
  return remap$(v, {
    recordUrl: "record_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CRMCompanyMatch$ {
  /** @deprecated use `CRMCompanyMatch$inboundSchema` instead. */
  export const inboundSchema = CRMCompanyMatch$inboundSchema;
  /** @deprecated use `CRMCompanyMatch$outboundSchema` instead. */
  export const outboundSchema = CRMCompanyMatch$outboundSchema;
  /** @deprecated use `CRMCompanyMatch$Outbound` instead. */
  export type Outbound = CRMCompanyMatch$Outbound;
}

export function crmCompanyMatchToJSON(
  crmCompanyMatch: CRMCompanyMatch,
): string {
  return JSON.stringify(CRMCompanyMatch$outboundSchema.parse(crmCompanyMatch));
}

export function crmCompanyMatchFromJSON(
  jsonString: string,
): SafeParseResult<CRMCompanyMatch, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CRMCompanyMatch$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CRMCompanyMatch' from JSON`,
  );
}
