// fathom-typescript/src/sdk/models/shared/crmcontactmatch.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type CRMContactMatch = {
  name: string;
  email: string;
  recordUrl: string;
};

/** @internal */
export const CRMContactMatch$inboundSchema: z.ZodType<
  CRMContactMatch,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  email: z.string(),
  record_url: z.string(),
}).transform((v) => {
  return remap$(v, {
    "record_url": "recordUrl",
  });
});

/** @internal */
export type CRMContactMatch$Outbound = {
  name: string;
  email: string;
  record_url: string;
};

/** @internal */
export const CRMContactMatch$outboundSchema: z.ZodType<
  CRMContactMatch$Outbound,
  z.ZodTypeDef,
  CRMContactMatch
> = z.object({
  name: z.string(),
  email: z.string(),
  recordUrl: z.string(),
}).transform((v) => {
  return remap$(v, {
    recordUrl: "record_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CRMContactMatch$ {
  /** @deprecated use `CRMContactMatch$inboundSchema` instead. */
  export const inboundSchema = CRMContactMatch$inboundSchema;
  /** @deprecated use `CRMContactMatch$outboundSchema` instead. */
  export const outboundSchema = CRMContactMatch$outboundSchema;
  /** @deprecated use `CRMContactMatch$Outbound` instead. */
  export type Outbound = CRMContactMatch$Outbound;
}

export function crmContactMatchToJSON(
  crmContactMatch: CRMContactMatch,
): string {
  return JSON.stringify(CRMContactMatch$outboundSchema.parse(crmContactMatch));
}

export function crmContactMatchFromJSON(
  jsonString: string,
): SafeParseResult<CRMContactMatch, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CRMContactMatch$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CRMContactMatch' from JSON`,
  );
}
