// fathom-typescript/src/sdk/models/shared/crmdealmatch.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type CRMDealMatch = {
  name: string;
  /**
   * The amount in dollars.
   */
  amount: number;
  recordUrl: string;
};

/** @internal */
export const CRMDealMatch$inboundSchema: z.ZodType<
  CRMDealMatch,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  amount: z.number(),
  record_url: z.string(),
}).transform((v) => {
  return remap$(v, {
    "record_url": "recordUrl",
  });
});

/** @internal */
export type CRMDealMatch$Outbound = {
  name: string;
  amount: number;
  record_url: string;
};

/** @internal */
export const CRMDealMatch$outboundSchema: z.ZodType<
  CRMDealMatch$Outbound,
  z.ZodTypeDef,
  CRMDealMatch
> = z.object({
  name: z.string(),
  amount: z.number(),
  recordUrl: z.string(),
}).transform((v) => {
  return remap$(v, {
    recordUrl: "record_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CRMDealMatch$ {
  /** @deprecated use `CRMDealMatch$inboundSchema` instead. */
  export const inboundSchema = CRMDealMatch$inboundSchema;
  /** @deprecated use `CRMDealMatch$outboundSchema` instead. */
  export const outboundSchema = CRMDealMatch$outboundSchema;
  /** @deprecated use `CRMDealMatch$Outbound` instead. */
  export type Outbound = CRMDealMatch$Outbound;
}

export function crmDealMatchToJSON(crmDealMatch: CRMDealMatch): string {
  return JSON.stringify(CRMDealMatch$outboundSchema.parse(crmDealMatch));
}

export function crmDealMatchFromJSON(
  jsonString: string,
): SafeParseResult<CRMDealMatch, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CRMDealMatch$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CRMDealMatch' from JSON`,
  );
}
