// fathom-typescript/src/sdk/models/shared/crmmatches.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  CRMCompanyMatch,
  CRMCompanyMatch$inboundSchema,
  CRMCompanyMatch$Outbound,
  CRMCompanyMatch$outboundSchema,
} from "./crmcompanymatch.js";
import {
  CRMContactMatch,
  CRMContactMatch$inboundSchema,
  CRMContactMatch$Outbound,
  CRMContactMatch$outboundSchema,
} from "./crmcontactmatch.js";
import {
  CRMDealMatch,
  CRMDealMatch$inboundSchema,
  CRMDealMatch$Outbound,
  CRMDealMatch$outboundSchema,
} from "./crmdealmatch.js";

/**
 * CRM data linked to the meeting. Only returns data from your or your team's linked CRM.
 *
 * @remarks
 * If no CRM is connected for the workspace, the `error` field will be populated.
 */
export type CRMMatches = {
  contacts?: Array<CRMContactMatch> | undefined;
  companies?: Array<CRMCompanyMatch> | undefined;
  deals?: Array<CRMDealMatch> | undefined;
  error?: string | null | undefined;
};

/** @internal */
export const CRMMatches$inboundSchema: z.ZodType<
  CRMMatches,
  z.ZodTypeDef,
  unknown
> = z.object({
  contacts: z.array(CRMContactMatch$inboundSchema).optional(),
  companies: z.array(CRMCompanyMatch$inboundSchema).optional(),
  deals: z.array(CRMDealMatch$inboundSchema).optional(),
  error: z.nullable(z.string()).optional(),
});

/** @internal */
export type CRMMatches$Outbound = {
  contacts?: Array<CRMContactMatch$Outbound> | undefined;
  companies?: Array<CRMCompanyMatch$Outbound> | undefined;
  deals?: Array<CRMDealMatch$Outbound> | undefined;
  error?: string | null | undefined;
};

/** @internal */
export const CRMMatches$outboundSchema: z.ZodType<
  CRMMatches$Outbound,
  z.ZodTypeDef,
  CRMMatches
> = z.object({
  contacts: z.array(CRMContactMatch$outboundSchema).optional(),
  companies: z.array(CRMCompanyMatch$outboundSchema).optional(),
  deals: z.array(CRMDealMatch$outboundSchema).optional(),
  error: z.nullable(z.string()).optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CRMMatches$ {
  /** @deprecated use `CRMMatches$inboundSchema` instead. */
  export const inboundSchema = CRMMatches$inboundSchema;
  /** @deprecated use `CRMMatches$outboundSchema` instead. */
  export const outboundSchema = CRMMatches$outboundSchema;
  /** @deprecated use `CRMMatches$Outbound` instead. */
  export type Outbound = CRMMatches$Outbound;
}

export function crmMatchesToJSON(crmMatches: CRMMatches): string {
  return JSON.stringify(CRMMatches$outboundSchema.parse(crmMatches));
}

export function crmMatchesFromJSON(
  jsonString: string,
): SafeParseResult<CRMMatches, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CRMMatches$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CRMMatches' from JSON`,
  );
}
