// fathom-typescript/src/sdk/models/shared/fathomuser.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type FathomUser = {
  name: string;
  email: string;
  emailDomain: string;
  team: string | null;
};

/** @internal */
export const FathomUser$inboundSchema: z.ZodType<
  FathomUser,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  email: z.string(),
  email_domain: z.string(),
  team: z.nullable(z.string()),
}).transform((v) => {
  return remap$(v, {
    "email_domain": "emailDomain",
  });
});

/** @internal */
export type FathomUser$Outbound = {
  name: string;
  email: string;
  email_domain: string;
  team: string | null;
};

/** @internal */
export const FathomUser$outboundSchema: z.ZodType<
  FathomUser$Outbound,
  z.ZodTypeDef,
  FathomUser
> = z.object({
  name: z.string(),
  email: z.string(),
  emailDomain: z.string(),
  team: z.nullable(z.string()),
}).transform((v) => {
  return remap$(v, {
    emailDomain: "email_domain",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace FathomUser$ {
  /** @deprecated use `FathomUser$inboundSchema` instead. */
  export const inboundSchema = FathomUser$inboundSchema;
  /** @deprecated use `FathomUser$outboundSchema` instead. */
  export const outboundSchema = FathomUser$outboundSchema;
  /** @deprecated use `FathomUser$Outbound` instead. */
  export type Outbound = FathomUser$Outbound;
}

export function fathomUserToJSON(fathomUser: FathomUser): string {
  return JSON.stringify(FathomUser$outboundSchema.parse(fathomUser));
}

export function fathomUserFromJSON(
  jsonString: string,
): SafeParseResult<FathomUser, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => FathomUser$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'FathomUser' from JSON`,
  );
}
