// fathom-typescript/src/sdk/models/shared/invitee.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type Invitee = {
  name: string | null;
  /**
   * **Coming soon!**
   *
   * @remarks
   */
  matchedSpeakerDisplayName?: string | null | undefined;
  email: string | null;
  emailDomain: string | null;
  isExternal: boolean;
};

/** @internal */
export const Invitee$inboundSchema: z.ZodType<Invitee, z.ZodTypeDef, unknown> =
  z.object({
    name: z.nullable(z.string()),
    matched_speaker_display_name: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()),
    email_domain: z.nullable(z.string()),
    is_external: z.boolean(),
  }).transform((v) => {
    return remap$(v, {
      "matched_speaker_display_name": "matchedSpeakerDisplayName",
      "email_domain": "emailDomain",
      "is_external": "isExternal",
    });
  });

/** @internal */
export type Invitee$Outbound = {
  name: string | null;
  matched_speaker_display_name?: string | null | undefined;
  email: string | null;
  email_domain: string | null;
  is_external: boolean;
};

/** @internal */
export const Invitee$outboundSchema: z.ZodType<
  Invitee$Outbound,
  z.ZodTypeDef,
  Invitee
> = z.object({
  name: z.nullable(z.string()),
  matchedSpeakerDisplayName: z.nullable(z.string()).optional(),
  email: z.nullable(z.string()),
  emailDomain: z.nullable(z.string()),
  isExternal: z.boolean(),
}).transform((v) => {
  return remap$(v, {
    matchedSpeakerDisplayName: "matched_speaker_display_name",
    emailDomain: "email_domain",
    isExternal: "is_external",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace Invitee$ {
  /** @deprecated use `Invitee$inboundSchema` instead. */
  export const inboundSchema = Invitee$inboundSchema;
  /** @deprecated use `Invitee$outboundSchema` instead. */
  export const outboundSchema = Invitee$outboundSchema;
  /** @deprecated use `Invitee$Outbound` instead. */
  export type Outbound = Invitee$Outbound;
}

export function inviteeToJSON(invitee: Invitee): string {
  return JSON.stringify(Invitee$outboundSchema.parse(invitee));
}

export function inviteeFromJSON(
  jsonString: string,
): SafeParseResult<Invitee, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => Invitee$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'Invitee' from JSON`,
  );
}
