// fathom-typescript/src/sdk/models/shared/meetingsummary.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type MeetingSummary = {
  templateName: string | null;
  /**
   * Always displayed in English.
   */
  markdownFormatted: string | null;
};

/** @internal */
export const MeetingSummary$inboundSchema: z.ZodType<
  MeetingSummary,
  z.ZodTypeDef,
  unknown
> = z.object({
  template_name: z.nullable(z.string()),
  markdown_formatted: z.nullable(z.string()),
}).transform((v) => {
  return remap$(v, {
    "template_name": "templateName",
    "markdown_formatted": "markdownFormatted",
  });
});

/** @internal */
export type MeetingSummary$Outbound = {
  template_name: string | null;
  markdown_formatted: string | null;
};

/** @internal */
export const MeetingSummary$outboundSchema: z.ZodType<
  MeetingSummary$Outbound,
  z.ZodTypeDef,
  MeetingSummary
> = z.object({
  templateName: z.nullable(z.string()),
  markdownFormatted: z.nullable(z.string()),
}).transform((v) => {
  return remap$(v, {
    templateName: "template_name",
    markdownFormatted: "markdown_formatted",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace MeetingSummary$ {
  /** @deprecated use `MeetingSummary$inboundSchema` instead. */
  export const inboundSchema = MeetingSummary$inboundSchema;
  /** @deprecated use `MeetingSummary$outboundSchema` instead. */
  export const outboundSchema = MeetingSummary$outboundSchema;
  /** @deprecated use `MeetingSummary$Outbound` instead. */
  export type Outbound = MeetingSummary$Outbound;
}

export function meetingSummaryToJSON(meetingSummary: MeetingSummary): string {
  return JSON.stringify(MeetingSummary$outboundSchema.parse(meetingSummary));
}

export function meetingSummaryFromJSON(
  jsonString: string,
): SafeParseResult<MeetingSummary, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => MeetingSummary$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'MeetingSummary' from JSON`,
  );
}
