// fathom-typescript/src/sdk/models/shared/meetinglistresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  Meeting,
  Meeting$inboundSchema,
  Meeting$Outbound,
  Meeting$outboundSchema,
} from "./meeting.js";

export type MeetingListResponse = {
  limit: number | null;
  nextCursor: string | null;
  items: Array<Meeting>;
};

/** @internal */
export const MeetingListResponse$inboundSchema: z.ZodType<
  MeetingListResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  limit: z.nullable(z.number().int()),
  next_cursor: z.nullable(z.string()),
  items: z.array(Meeting$inboundSchema),
}).transform((v) => {
  return remap$(v, {
    "next_cursor": "nextCursor",
  });
});

/** @internal */
export type MeetingListResponse$Outbound = {
  limit: number | null;
  next_cursor: string | null;
  items: Array<Meeting$Outbound>;
};

/** @internal */
export const MeetingListResponse$outboundSchema: z.ZodType<
  MeetingListResponse$Outbound,
  z.ZodTypeDef,
  MeetingListResponse
> = z.object({
  limit: z.nullable(z.number().int()),
  nextCursor: z.nullable(z.string()),
  items: z.array(Meeting$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    nextCursor: "next_cursor",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace MeetingListResponse$ {
  /** @deprecated use `MeetingListResponse$inboundSchema` instead. */
  export const inboundSchema = MeetingListResponse$inboundSchema;
  /** @deprecated use `MeetingListResponse$outboundSchema` instead. */
  export const outboundSchema = MeetingListResponse$outboundSchema;
  /** @deprecated use `MeetingListResponse$Outbound` instead. */
  export type Outbound = MeetingListResponse$Outbound;
}

export function meetingListResponseToJSON(
  meetingListResponse: MeetingListResponse,
): string {
  return JSON.stringify(
    MeetingListResponse$outboundSchema.parse(meetingListResponse),
  );
}

export function meetingListResponseFromJSON(
  jsonString: string,
): SafeParseResult<MeetingListResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => MeetingListResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'MeetingListResponse' from JSON`,
  );
}
