// fathom-typescript/src/sdk/models/shared/meeting.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { ClosedEnum } from "../../types/enums.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  ActionItem,
  ActionItem$inboundSchema,
  ActionItem$Outbound,
  ActionItem$outboundSchema,
} from "./actionitem.js";
import {
  CRMMatches,
  CRMMatches$inboundSchema,
  CRMMatches$Outbound,
  CRMMatches$outboundSchema,
} from "./crmmatches.js";
import {
  FathomUser,
  FathomUser$inboundSchema,
  FathomUser$Outbound,
  FathomUser$outboundSchema,
} from "./fathomuser.js";
import {
  Invitee,
  Invitee$inboundSchema,
  Invitee$Outbound,
  Invitee$outboundSchema,
} from "./invitee.js";
import {
  MeetingSummary,
  MeetingSummary$inboundSchema,
  MeetingSummary$Outbound,
  MeetingSummary$outboundSchema,
} from "./meetingsummary.js";
import {
  TranscriptItem,
  TranscriptItem$inboundSchema,
  TranscriptItem$Outbound,
  TranscriptItem$outboundSchema,
} from "./transcriptitem.js";

export const CalendarInviteesDomainsType = {
  OnlyInternal: "only_internal",
  OneOrMoreExternal: "one_or_more_external",
} as const;
export type CalendarInviteesDomainsType = ClosedEnum<
  typeof CalendarInviteesDomainsType
>;

export type Meeting = {
  title: string;
  /**
   * Calendar event title.
   */
  meetingTitle: string | null;
  /**
   * The ID of the meeting recording.
   */
  recordingId: number;
  url: string;
  shareUrl: string;
  createdAt: Date;
  scheduledStartTime: Date;
  scheduledEndTime: Date;
  recordingStartTime: Date;
  recordingEndTime: Date;
  calendarInviteesDomainsType: CalendarInviteesDomainsType;
  transcriptLanguage: string;
  transcript?: Array<TranscriptItem> | null | undefined;
  defaultSummary?: MeetingSummary | null | undefined;
  actionItems?: Array<ActionItem> | null | undefined;
  calendarInvitees: Array<Invitee>;
  recordedBy: FathomUser;
  /**
   * CRM data linked to the meeting. Only returns data from your or your team's linked CRM.
   *
   * @remarks
   * If no CRM is connected for the workspace, the `error` field will be populated.
   */
  crmMatches?: CRMMatches | null | undefined;
};

/** @internal */
export const CalendarInviteesDomainsType$inboundSchema: z.ZodNativeEnum<
  typeof CalendarInviteesDomainsType
> = z.nativeEnum(CalendarInviteesDomainsType);

/** @internal */
export const CalendarInviteesDomainsType$outboundSchema: z.ZodNativeEnum<
  typeof CalendarInviteesDomainsType
> = CalendarInviteesDomainsType$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CalendarInviteesDomainsType$ {
  /** @deprecated use `CalendarInviteesDomainsType$inboundSchema` instead. */
  export const inboundSchema = CalendarInviteesDomainsType$inboundSchema;
  /** @deprecated use `CalendarInviteesDomainsType$outboundSchema` instead. */
  export const outboundSchema = CalendarInviteesDomainsType$outboundSchema;
}

/** @internal */
export const Meeting$inboundSchema: z.ZodType<Meeting, z.ZodTypeDef, unknown> =
  z.object({
    title: z.string(),
    meeting_title: z.nullable(z.string()),
    recording_id: z.number().int(),
    url: z.string(),
    share_url: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v =>
      new Date(v)
    ),
    scheduled_start_time: z.string().datetime({ offset: true }).transform(v =>
      new Date(v)
    ),
    scheduled_end_time: z.string().datetime({ offset: true }).transform(v =>
      new Date(v)
    ),
    recording_start_time: z.string().datetime({ offset: true }).transform(v =>
      new Date(v)
    ),
    recording_end_time: z.string().datetime({ offset: true }).transform(v =>
      new Date(v)
    ),
    calendar_invitees_domains_type: CalendarInviteesDomainsType$inboundSchema,
    transcript_language: z.string(),
    transcript: z.nullable(z.array(TranscriptItem$inboundSchema)).optional(),
    default_summary: z.nullable(MeetingSummary$inboundSchema).optional(),
    action_items: z.nullable(z.array(ActionItem$inboundSchema)).optional(),
    calendar_invitees: z.array(Invitee$inboundSchema),
    recorded_by: FathomUser$inboundSchema,
    crm_matches: z.nullable(CRMMatches$inboundSchema).optional(),
  }).transform((v) => {
    return remap$(v, {
      "meeting_title": "meetingTitle",
      "recording_id": "recordingId",
      "share_url": "shareUrl",
      "created_at": "createdAt",
      "scheduled_start_time": "scheduledStartTime",
      "scheduled_end_time": "scheduledEndTime",
      "recording_start_time": "recordingStartTime",
      "recording_end_time": "recordingEndTime",
      "calendar_invitees_domains_type": "calendarInviteesDomainsType",
      "transcript_language": "transcriptLanguage",
      "default_summary": "defaultSummary",
      "action_items": "actionItems",
      "calendar_invitees": "calendarInvitees",
      "recorded_by": "recordedBy",
      "crm_matches": "crmMatches",
    });
  });

/** @internal */
export type Meeting$Outbound = {
  title: string;
  meeting_title: string | null;
  recording_id: number;
  url: string;
  share_url: string;
  created_at: string;
  scheduled_start_time: string;
  scheduled_end_time: string;
  recording_start_time: string;
  recording_end_time: string;
  calendar_invitees_domains_type: string;
  transcript_language: string;
  transcript?: Array<TranscriptItem$Outbound> | null | undefined;
  default_summary?: MeetingSummary$Outbound | null | undefined;
  action_items?: Array<ActionItem$Outbound> | null | undefined;
  calendar_invitees: Array<Invitee$Outbound>;
  recorded_by: FathomUser$Outbound;
  crm_matches?: CRMMatches$Outbound | null | undefined;
};

/** @internal */
export const Meeting$outboundSchema: z.ZodType<
  Meeting$Outbound,
  z.ZodTypeDef,
  Meeting
> = z.object({
  title: z.string(),
  meetingTitle: z.nullable(z.string()),
  recordingId: z.number().int(),
  url: z.string(),
  shareUrl: z.string(),
  createdAt: z.date().transform(v => v.toISOString()),
  scheduledStartTime: z.date().transform(v => v.toISOString()),
  scheduledEndTime: z.date().transform(v => v.toISOString()),
  recordingStartTime: z.date().transform(v => v.toISOString()),
  recordingEndTime: z.date().transform(v => v.toISOString()),
  calendarInviteesDomainsType: CalendarInviteesDomainsType$outboundSchema,
  transcriptLanguage: z.string(),
  transcript: z.nullable(z.array(TranscriptItem$outboundSchema)).optional(),
  defaultSummary: z.nullable(MeetingSummary$outboundSchema).optional(),
  actionItems: z.nullable(z.array(ActionItem$outboundSchema)).optional(),
  calendarInvitees: z.array(Invitee$outboundSchema),
  recordedBy: FathomUser$outboundSchema,
  crmMatches: z.nullable(CRMMatches$outboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    meetingTitle: "meeting_title",
    recordingId: "recording_id",
    shareUrl: "share_url",
    createdAt: "created_at",
    scheduledStartTime: "scheduled_start_time",
    scheduledEndTime: "scheduled_end_time",
    recordingStartTime: "recording_start_time",
    recordingEndTime: "recording_end_time",
    calendarInviteesDomainsType: "calendar_invitees_domains_type",
    transcriptLanguage: "transcript_language",
    defaultSummary: "default_summary",
    actionItems: "action_items",
    calendarInvitees: "calendar_invitees",
    recordedBy: "recorded_by",
    crmMatches: "crm_matches",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace Meeting$ {
  /** @deprecated use `Meeting$inboundSchema` instead. */
  export const inboundSchema = Meeting$inboundSchema;
  /** @deprecated use `Meeting$outboundSchema` instead. */
  export const outboundSchema = Meeting$outboundSchema;
  /** @deprecated use `Meeting$Outbound` instead. */
  export type Outbound = Meeting$Outbound;
}

export function meetingToJSON(meeting: Meeting): string {
  return JSON.stringify(Meeting$outboundSchema.parse(meeting));
}

export function meetingFromJSON(
  jsonString: string,
): SafeParseResult<Meeting, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => Meeting$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'Meeting' from JSON`,
  );
}
