// fathom-typescript/src/sdk/models/shared/teammemberlistresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  TeamMember,
  TeamMember$inboundSchema,
  TeamMember$Outbound,
  TeamMember$outboundSchema,
} from "./teammember.js";

export type TeamMemberListResponse = {
  limit: number;
  nextCursor: string | null;
  items: Array<TeamMember>;
};

/** @internal */
export const TeamMemberListResponse$inboundSchema: z.ZodType<
  TeamMemberListResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  limit: z.number().int(),
  next_cursor: z.nullable(z.string()),
  items: z.array(TeamMember$inboundSchema),
}).transform((v) => {
  return remap$(v, {
    "next_cursor": "nextCursor",
  });
});

/** @internal */
export type TeamMemberListResponse$Outbound = {
  limit: number;
  next_cursor: string | null;
  items: Array<TeamMember$Outbound>;
};

/** @internal */
export const TeamMemberListResponse$outboundSchema: z.ZodType<
  TeamMemberListResponse$Outbound,
  z.ZodTypeDef,
  TeamMemberListResponse
> = z.object({
  limit: z.number().int(),
  nextCursor: z.nullable(z.string()),
  items: z.array(TeamMember$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    nextCursor: "next_cursor",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TeamMemberListResponse$ {
  /** @deprecated use `TeamMemberListResponse$inboundSchema` instead. */
  export const inboundSchema = TeamMemberListResponse$inboundSchema;
  /** @deprecated use `TeamMemberListResponse$outboundSchema` instead. */
  export const outboundSchema = TeamMemberListResponse$outboundSchema;
  /** @deprecated use `TeamMemberListResponse$Outbound` instead. */
  export type Outbound = TeamMemberListResponse$Outbound;
}

export function teamMemberListResponseToJSON(
  teamMemberListResponse: TeamMemberListResponse,
): string {
  return JSON.stringify(
    TeamMemberListResponse$outboundSchema.parse(teamMemberListResponse),
  );
}

export function teamMemberListResponseFromJSON(
  jsonString: string,
): SafeParseResult<TeamMemberListResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => TeamMemberListResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'TeamMemberListResponse' from JSON`,
  );
}
