// fathom-typescript/src/sdk/models/shared/teammember.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type TeamMember = {
  name: string;
  email: string;
  createdAt: Date;
};

/** @internal */
export const TeamMember$inboundSchema: z.ZodType<
  TeamMember,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  email: z.string(),
  created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
  return remap$(v, {
    "created_at": "createdAt",
  });
});

/** @internal */
export type TeamMember$Outbound = {
  name: string;
  email: string;
  created_at: string;
};

/** @internal */
export const TeamMember$outboundSchema: z.ZodType<
  TeamMember$Outbound,
  z.ZodTypeDef,
  TeamMember
> = z.object({
  name: z.string(),
  email: z.string(),
  createdAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
  return remap$(v, {
    createdAt: "created_at",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TeamMember$ {
  /** @deprecated use `TeamMember$inboundSchema` instead. */
  export const inboundSchema = TeamMember$inboundSchema;
  /** @deprecated use `TeamMember$outboundSchema` instead. */
  export const outboundSchema = TeamMember$outboundSchema;
  /** @deprecated use `TeamMember$Outbound` instead. */
  export type Outbound = TeamMember$Outbound;
}

export function teamMemberToJSON(teamMember: TeamMember): string {
  return JSON.stringify(TeamMember$outboundSchema.parse(teamMember));
}

export function teamMemberFromJSON(
  jsonString: string,
): SafeParseResult<TeamMember, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => TeamMember$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'TeamMember' from JSON`,
  );
}
