// fathom-typescript/src/sdk/models/shared/teamlistresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  Team,
  Team$inboundSchema,
  Team$Outbound,
  Team$outboundSchema,
} from "./team.js";

export type TeamListResponse = {
  limit: number;
  nextCursor: string | null;
  items: Array<Team>;
};

/** @internal */
export const TeamListResponse$inboundSchema: z.ZodType<
  TeamListResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  limit: z.number().int(),
  next_cursor: z.nullable(z.string()),
  items: z.array(Team$inboundSchema),
}).transform((v) => {
  return remap$(v, {
    "next_cursor": "nextCursor",
  });
});

/** @internal */
export type TeamListResponse$Outbound = {
  limit: number;
  next_cursor: string | null;
  items: Array<Team$Outbound>;
};

/** @internal */
export const TeamListResponse$outboundSchema: z.ZodType<
  TeamListResponse$Outbound,
  z.ZodTypeDef,
  TeamListResponse
> = z.object({
  limit: z.number().int(),
  nextCursor: z.nullable(z.string()),
  items: z.array(Team$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    nextCursor: "next_cursor",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TeamListResponse$ {
  /** @deprecated use `TeamListResponse$inboundSchema` instead. */
  export const inboundSchema = TeamListResponse$inboundSchema;
  /** @deprecated use `TeamListResponse$outboundSchema` instead. */
  export const outboundSchema = TeamListResponse$outboundSchema;
  /** @deprecated use `TeamListResponse$Outbound` instead. */
  export type Outbound = TeamListResponse$Outbound;
}

export function teamListResponseToJSON(
  teamListResponse: TeamListResponse,
): string {
  return JSON.stringify(
    TeamListResponse$outboundSchema.parse(teamListResponse),
  );
}

export function teamListResponseFromJSON(
  jsonString: string,
): SafeParseResult<TeamListResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => TeamListResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'TeamListResponse' from JSON`,
  );
}
