// fathom-typescript/src/sdk/models/shared/transcriptitemspeaker.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type TranscriptItemSpeaker = {
  displayName: string;
  /**
   * **Coming soon!**
   *
   * @remarks
   */
  matchedCalendarInviteeEmail?: string | null | undefined;
};

/** @internal */
export const TranscriptItemSpeaker$inboundSchema: z.ZodType<
  TranscriptItemSpeaker,
  z.ZodTypeDef,
  unknown
> = z.object({
  display_name: z.string(),
  matched_calendar_invitee_email: z.nullable(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    "display_name": "displayName",
    "matched_calendar_invitee_email": "matchedCalendarInviteeEmail",
  });
});

/** @internal */
export type TranscriptItemSpeaker$Outbound = {
  display_name: string;
  matched_calendar_invitee_email?: string | null | undefined;
};

/** @internal */
export const TranscriptItemSpeaker$outboundSchema: z.ZodType<
  TranscriptItemSpeaker$Outbound,
  z.ZodTypeDef,
  TranscriptItemSpeaker
> = z.object({
  displayName: z.string(),
  matchedCalendarInviteeEmail: z.nullable(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    displayName: "display_name",
    matchedCalendarInviteeEmail: "matched_calendar_invitee_email",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TranscriptItemSpeaker$ {
  /** @deprecated use `TranscriptItemSpeaker$inboundSchema` instead. */
  export const inboundSchema = TranscriptItemSpeaker$inboundSchema;
  /** @deprecated use `TranscriptItemSpeaker$outboundSchema` instead. */
  export const outboundSchema = TranscriptItemSpeaker$outboundSchema;
  /** @deprecated use `TranscriptItemSpeaker$Outbound` instead. */
  export type Outbound = TranscriptItemSpeaker$Outbound;
}

export function transcriptItemSpeakerToJSON(
  transcriptItemSpeaker: TranscriptItemSpeaker,
): string {
  return JSON.stringify(
    TranscriptItemSpeaker$outboundSchema.parse(transcriptItemSpeaker),
  );
}

export function transcriptItemSpeakerFromJSON(
  jsonString: string,
): SafeParseResult<TranscriptItemSpeaker, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => TranscriptItemSpeaker$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'TranscriptItemSpeaker' from JSON`,
  );
}
