// fathom-typescript/src/sdk/models/shared/transcriptitem.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  TranscriptItemSpeaker,
  TranscriptItemSpeaker$inboundSchema,
  TranscriptItemSpeaker$Outbound,
  TranscriptItemSpeaker$outboundSchema,
} from "./transcriptitemspeaker.js";

export type TranscriptItem = {
  speaker: TranscriptItemSpeaker;
  text: string;
  /**
   * Timestamp relative to the start of the meeting (HH:MM:SS).
   */
  timestamp: string;
};

/** @internal */
export const TranscriptItem$inboundSchema: z.ZodType<
  TranscriptItem,
  z.ZodTypeDef,
  unknown
> = z.object({
  speaker: TranscriptItemSpeaker$inboundSchema,
  text: z.string(),
  timestamp: z.string(),
});

/** @internal */
export type TranscriptItem$Outbound = {
  speaker: TranscriptItemSpeaker$Outbound;
  text: string;
  timestamp: string;
};

/** @internal */
export const TranscriptItem$outboundSchema: z.ZodType<
  TranscriptItem$Outbound,
  z.ZodTypeDef,
  TranscriptItem
> = z.object({
  speaker: TranscriptItemSpeaker$outboundSchema,
  text: z.string(),
  timestamp: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TranscriptItem$ {
  /** @deprecated use `TranscriptItem$inboundSchema` instead. */
  export const inboundSchema = TranscriptItem$inboundSchema;
  /** @deprecated use `TranscriptItem$outboundSchema` instead. */
  export const outboundSchema = TranscriptItem$outboundSchema;
  /** @deprecated use `TranscriptItem$Outbound` instead. */
  export type Outbound = TranscriptItem$Outbound;
}

export function transcriptItemToJSON(transcriptItem: TranscriptItem): string {
  return JSON.stringify(TranscriptItem$outboundSchema.parse(transcriptItem));
}

export function transcriptItemFromJSON(
  jsonString: string,
): SafeParseResult<TranscriptItem, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => TranscriptItem$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'TranscriptItem' from JSON`,
  );
}
