// fathom-typescript/src/sdk/models/shared/webhook.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { ClosedEnum } from "../../types/enums.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export const TriggeredFor = {
  MyRecordings: "my_recordings",
  SharedExternalRecordings: "shared_external_recordings",
  MySharedWithTeamRecordings: "my_shared_with_team_recordings",
  SharedTeamRecordings: "shared_team_recordings",
} as const;
export type TriggeredFor = ClosedEnum<typeof TriggeredFor>;

export type Webhook = {
  id: string;
  url: string;
  /**
   * The secret used to verify the webhook signature.
   */
  secret: string;
  /**
   * The date and time the webhook was created in ISO 8601 format.
   */
  createdAt: Date;
  includeTranscript: boolean;
  includeCrmMatches: boolean;
  includeSummary: boolean;
  includeActionItems: boolean;
  triggeredFor: Array<TriggeredFor>;
};

/** @internal */
export const TriggeredFor$inboundSchema: z.ZodNativeEnum<typeof TriggeredFor> =
  z.nativeEnum(TriggeredFor);

/** @internal */
export const TriggeredFor$outboundSchema: z.ZodNativeEnum<typeof TriggeredFor> =
  TriggeredFor$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TriggeredFor$ {
  /** @deprecated use `TriggeredFor$inboundSchema` instead. */
  export const inboundSchema = TriggeredFor$inboundSchema;
  /** @deprecated use `TriggeredFor$outboundSchema` instead. */
  export const outboundSchema = TriggeredFor$outboundSchema;
}

/** @internal */
export const Webhook$inboundSchema: z.ZodType<Webhook, z.ZodTypeDef, unknown> =
  z.object({
    id: z.string(),
    url: z.string(),
    secret: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v =>
      new Date(v)
    ),
    include_transcript: z.boolean(),
    include_crm_matches: z.boolean(),
    include_summary: z.boolean(),
    include_action_items: z.boolean(),
    triggered_for: z.array(TriggeredFor$inboundSchema),
  }).transform((v) => {
    return remap$(v, {
      "created_at": "createdAt",
      "include_transcript": "includeTranscript",
      "include_crm_matches": "includeCrmMatches",
      "include_summary": "includeSummary",
      "include_action_items": "includeActionItems",
      "triggered_for": "triggeredFor",
    });
  });

/** @internal */
export type Webhook$Outbound = {
  id: string;
  url: string;
  secret: string;
  created_at: string;
  include_transcript: boolean;
  include_crm_matches: boolean;
  include_summary: boolean;
  include_action_items: boolean;
  triggered_for: Array<string>;
};

/** @internal */
export const Webhook$outboundSchema: z.ZodType<
  Webhook$Outbound,
  z.ZodTypeDef,
  Webhook
> = z.object({
  id: z.string(),
  url: z.string(),
  secret: z.string(),
  createdAt: z.date().transform(v => v.toISOString()),
  includeTranscript: z.boolean(),
  includeCrmMatches: z.boolean(),
  includeSummary: z.boolean(),
  includeActionItems: z.boolean(),
  triggeredFor: z.array(TriggeredFor$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    createdAt: "created_at",
    includeTranscript: "include_transcript",
    includeCrmMatches: "include_crm_matches",
    includeSummary: "include_summary",
    includeActionItems: "include_action_items",
    triggeredFor: "triggered_for",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace Webhook$ {
  /** @deprecated use `Webhook$inboundSchema` instead. */
  export const inboundSchema = Webhook$inboundSchema;
  /** @deprecated use `Webhook$outboundSchema` instead. */
  export const outboundSchema = Webhook$outboundSchema;
  /** @deprecated use `Webhook$Outbound` instead. */
  export type Outbound = Webhook$Outbound;
}

export function webhookToJSON(webhook: Webhook): string {
  return JSON.stringify(Webhook$outboundSchema.parse(webhook));
}

export function webhookFromJSON(
  jsonString: string,
): SafeParseResult<Webhook, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => Webhook$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'Webhook' from JSON`,
  );
}
