// fathom-typescript/dist/esm/sdk/models/shared/actionitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Assignee$inboundSchema, Assignee$outboundSchema, } from "./assignee.js";
/** @internal */
export const ActionItem$inboundSchema = z.object({
    description: z.string(),
    user_generated: z.boolean(),
    completed: z.boolean(),
    recording_timestamp: z.string(),
    recording_playback_url: z.string(),
    assignee: Assignee$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "user_generated": "userGenerated",
        "recording_timestamp": "recordingTimestamp",
        "recording_playback_url": "recordingPlaybackUrl",
    });
});
/** @internal */
export const ActionItem$outboundSchema = z.object({
    description: z.string(),
    userGenerated: z.boolean(),
    completed: z.boolean(),
    recordingTimestamp: z.string(),
    recordingPlaybackUrl: z.string(),
    assignee: Assignee$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        userGenerated: "user_generated",
        recordingTimestamp: "recording_timestamp",
        recordingPlaybackUrl: "recording_playback_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ActionItem$;
(function (ActionItem$) {
    /** @deprecated use `ActionItem$inboundSchema` instead. */
    ActionItem$.inboundSchema = ActionItem$inboundSchema;
    /** @deprecated use `ActionItem$outboundSchema` instead. */
    ActionItem$.outboundSchema = ActionItem$outboundSchema;
})(ActionItem$ || (ActionItem$ = {}));
export function actionItemToJSON(actionItem) {
    return JSON.stringify(ActionItem$outboundSchema.parse(actionItem));
}
export function actionItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActionItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActionItem' from JSON`);
}
