// fathom-typescript/dist/esm/sdk/models/shared/assignee.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const Assignee$inboundSchema = z.object({
    name: z.nullable(z.string()),
    email: z.nullable(z.string()),
    team: z.nullable(z.string()),
});
/** @internal */
export const Assignee$outboundSchema = z.object({
    name: z.nullable(z.string()),
    email: z.nullable(z.string()),
    team: z.nullable(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Assignee$;
(function (Assignee$) {
    /** @deprecated use `Assignee$inboundSchema` instead. */
    Assignee$.inboundSchema = Assignee$inboundSchema;
    /** @deprecated use `Assignee$outboundSchema` instead. */
    Assignee$.outboundSchema = Assignee$outboundSchema;
})(Assignee$ || (Assignee$ = {}));
export function assigneeToJSON(assignee) {
    return JSON.stringify(Assignee$outboundSchema.parse(assignee));
}
export function assigneeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Assignee$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Assignee' from JSON`);
}
