// fathom-typescript/dist/esm/sdk/models/shared/callbackresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const CallbackResponse$inboundSchema = z.object({
    destination_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "destination_url": "destinationUrl",
    });
});
/** @internal */
export const CallbackResponse$outboundSchema = z.object({
    destinationUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        destinationUrl: "destination_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CallbackResponse$;
(function (CallbackResponse$) {
    /** @deprecated use `CallbackResponse$inboundSchema` instead. */
    CallbackResponse$.inboundSchema = CallbackResponse$inboundSchema;
    /** @deprecated use `CallbackResponse$outboundSchema` instead. */
    CallbackResponse$.outboundSchema = CallbackResponse$outboundSchema;
})(CallbackResponse$ || (CallbackResponse$ = {}));
export function callbackResponseToJSON(callbackResponse) {
    return JSON.stringify(CallbackResponse$outboundSchema.parse(callbackResponse));
}
export function callbackResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CallbackResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CallbackResponse' from JSON`);
}
