// fathom-typescript/dist/esm/sdk/models/shared/crmcontactmatch.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const CRMContactMatch$inboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    record_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "record_url": "recordUrl",
    });
});
/** @internal */
export const CRMContactMatch$outboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    recordUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        recordUrl: "record_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CRMContactMatch$;
(function (CRMContactMatch$) {
    /** @deprecated use `CRMContactMatch$inboundSchema` instead. */
    CRMContactMatch$.inboundSchema = CRMContactMatch$inboundSchema;
    /** @deprecated use `CRMContactMatch$outboundSchema` instead. */
    CRMContactMatch$.outboundSchema = CRMContactMatch$outboundSchema;
})(CRMContactMatch$ || (CRMContactMatch$ = {}));
export function crmContactMatchToJSON(crmContactMatch) {
    return JSON.stringify(CRMContactMatch$outboundSchema.parse(crmContactMatch));
}
export function crmContactMatchFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CRMContactMatch$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CRMContactMatch' from JSON`);
}
