// fathom-typescript/dist/esm/sdk/models/shared/crmcompanymatch.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const CRMCompanyMatch$inboundSchema = z.object({
    name: z.string(),
    record_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "record_url": "recordUrl",
    });
});
/** @internal */
export const CRMCompanyMatch$outboundSchema = z.object({
    name: z.string(),
    recordUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        recordUrl: "record_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CRMCompanyMatch$;
(function (CRMCompanyMatch$) {
    /** @deprecated use `CRMCompanyMatch$inboundSchema` instead. */
    CRMCompanyMatch$.inboundSchema = CRMCompanyMatch$inboundSchema;
    /** @deprecated use `CRMCompanyMatch$outboundSchema` instead. */
    CRMCompanyMatch$.outboundSchema = CRMCompanyMatch$outboundSchema;
})(CRMCompanyMatch$ || (CRMCompanyMatch$ = {}));
export function crmCompanyMatchToJSON(crmCompanyMatch) {
    return JSON.stringify(CRMCompanyMatch$outboundSchema.parse(crmCompanyMatch));
}
export function crmCompanyMatchFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CRMCompanyMatch$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CRMCompanyMatch' from JSON`);
}
