// fathom-typescript/dist/esm/sdk/models/shared/crmdealmatch.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const CRMDealMatch$inboundSchema = z.object({
    name: z.string(),
    amount: z.number(),
    record_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "record_url": "recordUrl",
    });
});
/** @internal */
export const CRMDealMatch$outboundSchema = z.object({
    name: z.string(),
    amount: z.number(),
    recordUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        recordUrl: "record_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CRMDealMatch$;
(function (CRMDealMatch$) {
    /** @deprecated use `CRMDealMatch$inboundSchema` instead. */
    CRMDealMatch$.inboundSchema = CRMDealMatch$inboundSchema;
    /** @deprecated use `CRMDealMatch$outboundSchema` instead. */
    CRMDealMatch$.outboundSchema = CRMDealMatch$outboundSchema;
})(CRMDealMatch$ || (CRMDealMatch$ = {}));
export function crmDealMatchToJSON(crmDealMatch) {
    return JSON.stringify(CRMDealMatch$outboundSchema.parse(crmDealMatch));
}
export function crmDealMatchFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CRMDealMatch$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CRMDealMatch' from JSON`);
}
