// fathom-typescript/dist/esm/sdk/models/shared/crmmatches.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../../lib/schemas.js";
import { CRMCompanyMatch$inboundSchema, CRMCompanyMatch$outboundSchema, } from "./crmcompanymatch.js";
import { CRMContactMatch$inboundSchema, CRMContactMatch$outboundSchema, } from "./crmcontactmatch.js";
import { CRMDealMatch$inboundSchema, CRMDealMatch$outboundSchema, } from "./crmdealmatch.js";
/** @internal */
export const CRMMatches$inboundSchema = z.object({
    contacts: z.array(CRMContactMatch$inboundSchema).optional(),
    companies: z.array(CRMCompanyMatch$inboundSchema).optional(),
    deals: z.array(CRMDealMatch$inboundSchema).optional(),
    error: z.nullable(z.string()).optional(),
});
/** @internal */
export const CRMMatches$outboundSchema = z.object({
    contacts: z.array(CRMContactMatch$outboundSchema).optional(),
    companies: z.array(CRMCompanyMatch$outboundSchema).optional(),
    deals: z.array(CRMDealMatch$outboundSchema).optional(),
    error: z.nullable(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CRMMatches$;
(function (CRMMatches$) {
    /** @deprecated use `CRMMatches$inboundSchema` instead. */
    CRMMatches$.inboundSchema = CRMMatches$inboundSchema;
    /** @deprecated use `CRMMatches$outboundSchema` instead. */
    CRMMatches$.outboundSchema = CRMMatches$outboundSchema;
})(CRMMatches$ || (CRMMatches$ = {}));
export function crmMatchesToJSON(crmMatches) {
    return JSON.stringify(CRMMatches$outboundSchema.parse(crmMatches));
}
export function crmMatchesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CRMMatches$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CRMMatches' from JSON`);
}
