// fathom-typescript/dist/esm/sdk/models/shared/fathomuser.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const FathomUser$inboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    email_domain: z.string(),
    team: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "email_domain": "emailDomain",
    });
});
/** @internal */
export const FathomUser$outboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    emailDomain: z.string(),
    team: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        emailDomain: "email_domain",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FathomUser$;
(function (FathomUser$) {
    /** @deprecated use `FathomUser$inboundSchema` instead. */
    FathomUser$.inboundSchema = FathomUser$inboundSchema;
    /** @deprecated use `FathomUser$outboundSchema` instead. */
    FathomUser$.outboundSchema = FathomUser$outboundSchema;
})(FathomUser$ || (FathomUser$ = {}));
export function fathomUserToJSON(fathomUser) {
    return JSON.stringify(FathomUser$outboundSchema.parse(fathomUser));
}
export function fathomUserFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FathomUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FathomUser' from JSON`);
}
