// fathom-typescript/dist/esm/sdk/models/shared/invitee.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const Invitee$inboundSchema = z.object({
    name: z.nullable(z.string()),
    matched_speaker_display_name: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()),
    email_domain: z.nullable(z.string()),
    is_external: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "matched_speaker_display_name": "matchedSpeakerDisplayName",
        "email_domain": "emailDomain",
        "is_external": "isExternal",
    });
});
/** @internal */
export const Invitee$outboundSchema = z.object({
    name: z.nullable(z.string()),
    matchedSpeakerDisplayName: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()),
    emailDomain: z.nullable(z.string()),
    isExternal: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        matchedSpeakerDisplayName: "matched_speaker_display_name",
        emailDomain: "email_domain",
        isExternal: "is_external",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Invitee$;
(function (Invitee$) {
    /** @deprecated use `Invitee$inboundSchema` instead. */
    Invitee$.inboundSchema = Invitee$inboundSchema;
    /** @deprecated use `Invitee$outboundSchema` instead. */
    Invitee$.outboundSchema = Invitee$outboundSchema;
})(Invitee$ || (Invitee$ = {}));
export function inviteeToJSON(invitee) {
    return JSON.stringify(Invitee$outboundSchema.parse(invitee));
}
export function inviteeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Invitee$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Invitee' from JSON`);
}
