// fathom-typescript/dist/esm/sdk/models/shared/meeting.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { ActionItem$inboundSchema, ActionItem$outboundSchema, } from "./actionitem.js";
import { CRMMatches$inboundSchema, CRMMatches$outboundSchema, } from "./crmmatches.js";
import { FathomUser$inboundSchema, FathomUser$outboundSchema, } from "./fathomuser.js";
import { Invitee$inboundSchema, Invitee$outboundSchema, } from "./invitee.js";
import { MeetingSummary$inboundSchema, MeetingSummary$outboundSchema, } from "./meetingsummary.js";
import { TranscriptItem$inboundSchema, TranscriptItem$outboundSchema, } from "./transcriptitem.js";
export const CalendarInviteesDomainsType = {
    OnlyInternal: "only_internal",
    OneOrMoreExternal: "one_or_more_external",
};
/** @internal */
export const CalendarInviteesDomainsType$inboundSchema = z.nativeEnum(CalendarInviteesDomainsType);
/** @internal */
export const CalendarInviteesDomainsType$outboundSchema = CalendarInviteesDomainsType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CalendarInviteesDomainsType$;
(function (CalendarInviteesDomainsType$) {
    /** @deprecated use `CalendarInviteesDomainsType$inboundSchema` instead. */
    CalendarInviteesDomainsType$.inboundSchema = CalendarInviteesDomainsType$inboundSchema;
    /** @deprecated use `CalendarInviteesDomainsType$outboundSchema` instead. */
    CalendarInviteesDomainsType$.outboundSchema = CalendarInviteesDomainsType$outboundSchema;
})(CalendarInviteesDomainsType$ || (CalendarInviteesDomainsType$ = {}));
/** @internal */
export const Meeting$inboundSchema = z.object({
    title: z.string(),
    meeting_title: z.nullable(z.string()),
    recording_id: z.number().int(),
    url: z.string(),
    share_url: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    scheduled_start_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    scheduled_end_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    recording_start_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    recording_end_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    calendar_invitees_domains_type: CalendarInviteesDomainsType$inboundSchema,
    transcript_language: z.string(),
    transcript: z.nullable(z.array(TranscriptItem$inboundSchema)).optional(),
    default_summary: z.nullable(MeetingSummary$inboundSchema).optional(),
    action_items: z.nullable(z.array(ActionItem$inboundSchema)).optional(),
    calendar_invitees: z.array(Invitee$inboundSchema),
    recorded_by: FathomUser$inboundSchema,
    crm_matches: z.nullable(CRMMatches$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "meeting_title": "meetingTitle",
        "recording_id": "recordingId",
        "share_url": "shareUrl",
        "created_at": "createdAt",
        "scheduled_start_time": "scheduledStartTime",
        "scheduled_end_time": "scheduledEndTime",
        "recording_start_time": "recordingStartTime",
        "recording_end_time": "recordingEndTime",
        "calendar_invitees_domains_type": "calendarInviteesDomainsType",
        "transcript_language": "transcriptLanguage",
        "default_summary": "defaultSummary",
        "action_items": "actionItems",
        "calendar_invitees": "calendarInvitees",
        "recorded_by": "recordedBy",
        "crm_matches": "crmMatches",
    });
});
/** @internal */
export const Meeting$outboundSchema = z.object({
    title: z.string(),
    meetingTitle: z.nullable(z.string()),
    recordingId: z.number().int(),
    url: z.string(),
    shareUrl: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    scheduledStartTime: z.date().transform(v => v.toISOString()),
    scheduledEndTime: z.date().transform(v => v.toISOString()),
    recordingStartTime: z.date().transform(v => v.toISOString()),
    recordingEndTime: z.date().transform(v => v.toISOString()),
    calendarInviteesDomainsType: CalendarInviteesDomainsType$outboundSchema,
    transcriptLanguage: z.string(),
    transcript: z.nullable(z.array(TranscriptItem$outboundSchema)).optional(),
    defaultSummary: z.nullable(MeetingSummary$outboundSchema).optional(),
    actionItems: z.nullable(z.array(ActionItem$outboundSchema)).optional(),
    calendarInvitees: z.array(Invitee$outboundSchema),
    recordedBy: FathomUser$outboundSchema,
    crmMatches: z.nullable(CRMMatches$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        meetingTitle: "meeting_title",
        recordingId: "recording_id",
        shareUrl: "share_url",
        createdAt: "created_at",
        scheduledStartTime: "scheduled_start_time",
        scheduledEndTime: "scheduled_end_time",
        recordingStartTime: "recording_start_time",
        recordingEndTime: "recording_end_time",
        calendarInviteesDomainsType: "calendar_invitees_domains_type",
        transcriptLanguage: "transcript_language",
        defaultSummary: "default_summary",
        actionItems: "action_items",
        calendarInvitees: "calendar_invitees",
        recordedBy: "recorded_by",
        crmMatches: "crm_matches",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Meeting$;
(function (Meeting$) {
    /** @deprecated use `Meeting$inboundSchema` instead. */
    Meeting$.inboundSchema = Meeting$inboundSchema;
    /** @deprecated use `Meeting$outboundSchema` instead. */
    Meeting$.outboundSchema = Meeting$outboundSchema;
})(Meeting$ || (Meeting$ = {}));
export function meetingToJSON(meeting) {
    return JSON.stringify(Meeting$outboundSchema.parse(meeting));
}
export function meetingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Meeting$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Meeting' from JSON`);
}
