// fathom-typescript/dist/esm/sdk/models/shared/meetinglistresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Meeting$inboundSchema, Meeting$outboundSchema, } from "./meeting.js";
/** @internal */
export const MeetingListResponse$inboundSchema = z.object({
    limit: z.nullable(z.number().int()),
    next_cursor: z.nullable(z.string()),
    items: z.array(Meeting$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "next_cursor": "nextCursor",
    });
});
/** @internal */
export const MeetingListResponse$outboundSchema = z.object({
    limit: z.nullable(z.number().int()),
    nextCursor: z.nullable(z.string()),
    items: z.array(Meeting$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        nextCursor: "next_cursor",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MeetingListResponse$;
(function (MeetingListResponse$) {
    /** @deprecated use `MeetingListResponse$inboundSchema` instead. */
    MeetingListResponse$.inboundSchema = MeetingListResponse$inboundSchema;
    /** @deprecated use `MeetingListResponse$outboundSchema` instead. */
    MeetingListResponse$.outboundSchema = MeetingListResponse$outboundSchema;
})(MeetingListResponse$ || (MeetingListResponse$ = {}));
export function meetingListResponseToJSON(meetingListResponse) {
    return JSON.stringify(MeetingListResponse$outboundSchema.parse(meetingListResponse));
}
export function meetingListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MeetingListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeetingListResponse' from JSON`);
}
