// fathom-typescript/dist/esm/sdk/models/shared/meetingsummary.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const MeetingSummary$inboundSchema = z.object({
    template_name: z.nullable(z.string()),
    markdown_formatted: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        "template_name": "templateName",
        "markdown_formatted": "markdownFormatted",
    });
});
/** @internal */
export const MeetingSummary$outboundSchema = z.object({
    templateName: z.nullable(z.string()),
    markdownFormatted: z.nullable(z.string()),
}).transform((v) => {
    return remap$(v, {
        templateName: "template_name",
        markdownFormatted: "markdown_formatted",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MeetingSummary$;
(function (MeetingSummary$) {
    /** @deprecated use `MeetingSummary$inboundSchema` instead. */
    MeetingSummary$.inboundSchema = MeetingSummary$inboundSchema;
    /** @deprecated use `MeetingSummary$outboundSchema` instead. */
    MeetingSummary$.outboundSchema = MeetingSummary$outboundSchema;
})(MeetingSummary$ || (MeetingSummary$ = {}));
export function meetingSummaryToJSON(meetingSummary) {
    return JSON.stringify(MeetingSummary$outboundSchema.parse(meetingSummary));
}
export function meetingSummaryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MeetingSummary$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeetingSummary' from JSON`);
}
