// fathom-typescript/dist/esm/sdk/models/shared/teamlistresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Team$inboundSchema, Team$outboundSchema, } from "./team.js";
/** @internal */
export const TeamListResponse$inboundSchema = z.object({
    limit: z.number().int(),
    next_cursor: z.nullable(z.string()),
    items: z.array(Team$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "next_cursor": "nextCursor",
    });
});
/** @internal */
export const TeamListResponse$outboundSchema = z.object({
    limit: z.number().int(),
    nextCursor: z.nullable(z.string()),
    items: z.array(Team$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        nextCursor: "next_cursor",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TeamListResponse$;
(function (TeamListResponse$) {
    /** @deprecated use `TeamListResponse$inboundSchema` instead. */
    TeamListResponse$.inboundSchema = TeamListResponse$inboundSchema;
    /** @deprecated use `TeamListResponse$outboundSchema` instead. */
    TeamListResponse$.outboundSchema = TeamListResponse$outboundSchema;
})(TeamListResponse$ || (TeamListResponse$ = {}));
export function teamListResponseToJSON(teamListResponse) {
    return JSON.stringify(TeamListResponse$outboundSchema.parse(teamListResponse));
}
export function teamListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TeamListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TeamListResponse' from JSON`);
}
