// fathom-typescript/dist/esm/sdk/models/shared/teammember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const TeamMember$inboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
    });
});
/** @internal */
export const TeamMember$outboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TeamMember$;
(function (TeamMember$) {
    /** @deprecated use `TeamMember$inboundSchema` instead. */
    TeamMember$.inboundSchema = TeamMember$inboundSchema;
    /** @deprecated use `TeamMember$outboundSchema` instead. */
    TeamMember$.outboundSchema = TeamMember$outboundSchema;
})(TeamMember$ || (TeamMember$ = {}));
export function teamMemberToJSON(teamMember) {
    return JSON.stringify(TeamMember$outboundSchema.parse(teamMember));
}
export function teamMemberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TeamMember$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TeamMember' from JSON`);
}
