// fathom-typescript/dist/esm/sdk/models/shared/team.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const Team$inboundSchema = z
    .object({
    name: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
    });
});
/** @internal */
export const Team$outboundSchema = z.object({
    name: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Team$;
(function (Team$) {
    /** @deprecated use `Team$inboundSchema` instead. */
    Team$.inboundSchema = Team$inboundSchema;
    /** @deprecated use `Team$outboundSchema` instead. */
    Team$.outboundSchema = Team$outboundSchema;
})(Team$ || (Team$ = {}));
export function teamToJSON(team) {
    return JSON.stringify(Team$outboundSchema.parse(team));
}
export function teamFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Team$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Team' from JSON`);
}
