// fathom-typescript/dist/esm/sdk/models/shared/teammemberlistresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { TeamMember$inboundSchema, TeamMember$outboundSchema, } from "./teammember.js";
/** @internal */
export const TeamMemberListResponse$inboundSchema = z.object({
    limit: z.number().int(),
    next_cursor: z.nullable(z.string()),
    items: z.array(TeamMember$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "next_cursor": "nextCursor",
    });
});
/** @internal */
export const TeamMemberListResponse$outboundSchema = z.object({
    limit: z.number().int(),
    nextCursor: z.nullable(z.string()),
    items: z.array(TeamMember$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        nextCursor: "next_cursor",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TeamMemberListResponse$;
(function (TeamMemberListResponse$) {
    /** @deprecated use `TeamMemberListResponse$inboundSchema` instead. */
    TeamMemberListResponse$.inboundSchema = TeamMemberListResponse$inboundSchema;
    /** @deprecated use `TeamMemberListResponse$outboundSchema` instead. */
    TeamMemberListResponse$.outboundSchema = TeamMemberListResponse$outboundSchema;
})(TeamMemberListResponse$ || (TeamMemberListResponse$ = {}));
export function teamMemberListResponseToJSON(teamMemberListResponse) {
    return JSON.stringify(TeamMemberListResponse$outboundSchema.parse(teamMemberListResponse));
}
export function teamMemberListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TeamMemberListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TeamMemberListResponse' from JSON`);
}
