// fathom-typescript/dist/esm/sdk/models/shared/transcriptitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../../lib/schemas.js";
import { TranscriptItemSpeaker$inboundSchema, TranscriptItemSpeaker$outboundSchema, } from "./transcriptitemspeaker.js";
/** @internal */
export const TranscriptItem$inboundSchema = z.object({
    speaker: TranscriptItemSpeaker$inboundSchema,
    text: z.string(),
    timestamp: z.string(),
});
/** @internal */
export const TranscriptItem$outboundSchema = z.object({
    speaker: TranscriptItemSpeaker$outboundSchema,
    text: z.string(),
    timestamp: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TranscriptItem$;
(function (TranscriptItem$) {
    /** @deprecated use `TranscriptItem$inboundSchema` instead. */
    TranscriptItem$.inboundSchema = TranscriptItem$inboundSchema;
    /** @deprecated use `TranscriptItem$outboundSchema` instead. */
    TranscriptItem$.outboundSchema = TranscriptItem$outboundSchema;
})(TranscriptItem$ || (TranscriptItem$ = {}));
export function transcriptItemToJSON(transcriptItem) {
    return JSON.stringify(TranscriptItem$outboundSchema.parse(transcriptItem));
}
export function transcriptItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TranscriptItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TranscriptItem' from JSON`);
}
