// fathom-typescript/dist/esm/sdk/models/shared/transcriptitemspeaker.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
/** @internal */
export const TranscriptItemSpeaker$inboundSchema = z.object({
    display_name: z.string(),
    matched_calendar_invitee_email: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "display_name": "displayName",
        "matched_calendar_invitee_email": "matchedCalendarInviteeEmail",
    });
});
/** @internal */
export const TranscriptItemSpeaker$outboundSchema = z.object({
    displayName: z.string(),
    matchedCalendarInviteeEmail: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        displayName: "display_name",
        matchedCalendarInviteeEmail: "matched_calendar_invitee_email",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TranscriptItemSpeaker$;
(function (TranscriptItemSpeaker$) {
    /** @deprecated use `TranscriptItemSpeaker$inboundSchema` instead. */
    TranscriptItemSpeaker$.inboundSchema = TranscriptItemSpeaker$inboundSchema;
    /** @deprecated use `TranscriptItemSpeaker$outboundSchema` instead. */
    TranscriptItemSpeaker$.outboundSchema = TranscriptItemSpeaker$outboundSchema;
})(TranscriptItemSpeaker$ || (TranscriptItemSpeaker$ = {}));
export function transcriptItemSpeakerToJSON(transcriptItemSpeaker) {
    return JSON.stringify(TranscriptItemSpeaker$outboundSchema.parse(transcriptItemSpeaker));
}
export function transcriptItemSpeakerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TranscriptItemSpeaker$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TranscriptItemSpeaker' from JSON`);
}
