// fathom-typescript/dist/esm/sdk/models/shared/webhook.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
export const TriggeredFor = {
    MyRecordings: "my_recordings",
    SharedExternalRecordings: "shared_external_recordings",
    MySharedWithTeamRecordings: "my_shared_with_team_recordings",
    SharedTeamRecordings: "shared_team_recordings",
};
/** @internal */
export const TriggeredFor$inboundSchema = z.nativeEnum(TriggeredFor);
/** @internal */
export const TriggeredFor$outboundSchema = TriggeredFor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TriggeredFor$;
(function (TriggeredFor$) {
    /** @deprecated use `TriggeredFor$inboundSchema` instead. */
    TriggeredFor$.inboundSchema = TriggeredFor$inboundSchema;
    /** @deprecated use `TriggeredFor$outboundSchema` instead. */
    TriggeredFor$.outboundSchema = TriggeredFor$outboundSchema;
})(TriggeredFor$ || (TriggeredFor$ = {}));
/** @internal */
export const Webhook$inboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    secret: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    include_transcript: z.boolean(),
    include_crm_matches: z.boolean(),
    include_summary: z.boolean(),
    include_action_items: z.boolean(),
    triggered_for: z.array(TriggeredFor$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "include_transcript": "includeTranscript",
        "include_crm_matches": "includeCrmMatches",
        "include_summary": "includeSummary",
        "include_action_items": "includeActionItems",
        "triggered_for": "triggeredFor",
    });
});
/** @internal */
export const Webhook$outboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    secret: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    includeTranscript: z.boolean(),
    includeCrmMatches: z.boolean(),
    includeSummary: z.boolean(),
    includeActionItems: z.boolean(),
    triggeredFor: z.array(TriggeredFor$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        includeTranscript: "include_transcript",
        includeCrmMatches: "include_crm_matches",
        includeSummary: "include_summary",
        includeActionItems: "include_action_items",
        triggeredFor: "triggered_for",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Webhook$;
(function (Webhook$) {
    /** @deprecated use `Webhook$inboundSchema` instead. */
    Webhook$.inboundSchema = Webhook$inboundSchema;
    /** @deprecated use `Webhook$outboundSchema` instead. */
    Webhook$.outboundSchema = Webhook$outboundSchema;
})(Webhook$ || (Webhook$ = {}));
export function webhookToJSON(webhook) {
    return JSON.stringify(Webhook$outboundSchema.parse(webhook));
}
export function webhookFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Webhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Webhook' from JSON`);
}
