// fathom-typescript/dist/commonjs/sdk/models/shared/actionitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionItem$ = exports.ActionItem$outboundSchema = exports.ActionItem$inboundSchema = void 0;
exports.actionItemToJSON = actionItemToJSON;
exports.actionItemFromJSON = actionItemFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const assignee_js_1 = require("./assignee.js");
/** @internal */
exports.ActionItem$inboundSchema = z.object({
    description: z.string(),
    user_generated: z.boolean(),
    completed: z.boolean(),
    recording_timestamp: z.string(),
    recording_playback_url: z.string(),
    assignee: assignee_js_1.Assignee$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "user_generated": "userGenerated",
        "recording_timestamp": "recordingTimestamp",
        "recording_playback_url": "recordingPlaybackUrl",
    });
});
/** @internal */
exports.ActionItem$outboundSchema = z.object({
    description: z.string(),
    userGenerated: z.boolean(),
    completed: z.boolean(),
    recordingTimestamp: z.string(),
    recordingPlaybackUrl: z.string(),
    assignee: assignee_js_1.Assignee$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userGenerated: "user_generated",
        recordingTimestamp: "recording_timestamp",
        recordingPlaybackUrl: "recording_playback_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ActionItem$;
(function (ActionItem$) {
    /** @deprecated use `ActionItem$inboundSchema` instead. */
    ActionItem$.inboundSchema = exports.ActionItem$inboundSchema;
    /** @deprecated use `ActionItem$outboundSchema` instead. */
    ActionItem$.outboundSchema = exports.ActionItem$outboundSchema;
})(ActionItem$ || (exports.ActionItem$ = ActionItem$ = {}));
function actionItemToJSON(actionItem) {
    return JSON.stringify(exports.ActionItem$outboundSchema.parse(actionItem));
}
function actionItemFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ActionItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActionItem' from JSON`);
}
