// fathom-typescript/dist/commonjs/sdk/models/shared/crmmatches.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CRMMatches$ = exports.CRMMatches$outboundSchema = exports.CRMMatches$inboundSchema = void 0;
exports.crmMatchesToJSON = crmMatchesToJSON;
exports.crmMatchesFromJSON = crmMatchesFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../../lib/schemas.js");
const crmcompanymatch_js_1 = require("./crmcompanymatch.js");
const crmcontactmatch_js_1 = require("./crmcontactmatch.js");
const crmdealmatch_js_1 = require("./crmdealmatch.js");
/** @internal */
exports.CRMMatches$inboundSchema = z.object({
    contacts: z.array(crmcontactmatch_js_1.CRMContactMatch$inboundSchema).optional(),
    companies: z.array(crmcompanymatch_js_1.CRMCompanyMatch$inboundSchema).optional(),
    deals: z.array(crmdealmatch_js_1.CRMDealMatch$inboundSchema).optional(),
    error: z.nullable(z.string()).optional(),
});
/** @internal */
exports.CRMMatches$outboundSchema = z.object({
    contacts: z.array(crmcontactmatch_js_1.CRMContactMatch$outboundSchema).optional(),
    companies: z.array(crmcompanymatch_js_1.CRMCompanyMatch$outboundSchema).optional(),
    deals: z.array(crmdealmatch_js_1.CRMDealMatch$outboundSchema).optional(),
    error: z.nullable(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CRMMatches$;
(function (CRMMatches$) {
    /** @deprecated use `CRMMatches$inboundSchema` instead. */
    CRMMatches$.inboundSchema = exports.CRMMatches$inboundSchema;
    /** @deprecated use `CRMMatches$outboundSchema` instead. */
    CRMMatches$.outboundSchema = exports.CRMMatches$outboundSchema;
})(CRMMatches$ || (exports.CRMMatches$ = CRMMatches$ = {}));
function crmMatchesToJSON(crmMatches) {
    return JSON.stringify(exports.CRMMatches$outboundSchema.parse(crmMatches));
}
function crmMatchesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CRMMatches$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CRMMatches' from JSON`);
}
