// fathom-typescript/dist/commonjs/sdk/models/shared/fathomuser.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FathomUser$ = exports.FathomUser$outboundSchema = exports.FathomUser$inboundSchema = void 0;
exports.fathomUserToJSON = fathomUserToJSON;
exports.fathomUserFromJSON = fathomUserFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.FathomUser$inboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    email_domain: z.string(),
    team: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "email_domain": "emailDomain",
    });
});
/** @internal */
exports.FathomUser$outboundSchema = z.object({
    name: z.string(),
    email: z.string(),
    emailDomain: z.string(),
    team: z.nullable(z.string()),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        emailDomain: "email_domain",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FathomUser$;
(function (FathomUser$) {
    /** @deprecated use `FathomUser$inboundSchema` instead. */
    FathomUser$.inboundSchema = exports.FathomUser$inboundSchema;
    /** @deprecated use `FathomUser$outboundSchema` instead. */
    FathomUser$.outboundSchema = exports.FathomUser$outboundSchema;
})(FathomUser$ || (exports.FathomUser$ = FathomUser$ = {}));
function fathomUserToJSON(fathomUser) {
    return JSON.stringify(exports.FathomUser$outboundSchema.parse(fathomUser));
}
function fathomUserFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FathomUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FathomUser' from JSON`);
}
