// fathom-typescript/dist/commonjs/sdk/models/shared/invitee.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Invitee$ = exports.Invitee$outboundSchema = exports.Invitee$inboundSchema = void 0;
exports.inviteeToJSON = inviteeToJSON;
exports.inviteeFromJSON = inviteeFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.Invitee$inboundSchema = z.object({
    name: z.nullable(z.string()),
    matched_speaker_display_name: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()),
    email_domain: z.nullable(z.string()),
    is_external: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "matched_speaker_display_name": "matchedSpeakerDisplayName",
        "email_domain": "emailDomain",
        "is_external": "isExternal",
    });
});
/** @internal */
exports.Invitee$outboundSchema = z.object({
    name: z.nullable(z.string()),
    matchedSpeakerDisplayName: z.nullable(z.string()).optional(),
    email: z.nullable(z.string()),
    emailDomain: z.nullable(z.string()),
    isExternal: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        matchedSpeakerDisplayName: "matched_speaker_display_name",
        emailDomain: "email_domain",
        isExternal: "is_external",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Invitee$;
(function (Invitee$) {
    /** @deprecated use `Invitee$inboundSchema` instead. */
    Invitee$.inboundSchema = exports.Invitee$inboundSchema;
    /** @deprecated use `Invitee$outboundSchema` instead. */
    Invitee$.outboundSchema = exports.Invitee$outboundSchema;
})(Invitee$ || (exports.Invitee$ = Invitee$ = {}));
function inviteeToJSON(invitee) {
    return JSON.stringify(exports.Invitee$outboundSchema.parse(invitee));
}
function inviteeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Invitee$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Invitee' from JSON`);
}
