// fathom-typescript/dist/commonjs/sdk/models/shared/meeting.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Meeting$ = exports.Meeting$outboundSchema = exports.Meeting$inboundSchema = exports.CalendarInviteesDomainsType$ = exports.CalendarInviteesDomainsType$outboundSchema = exports.CalendarInviteesDomainsType$inboundSchema = exports.CalendarInviteesDomainsType = void 0;
exports.meetingToJSON = meetingToJSON;
exports.meetingFromJSON = meetingFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const actionitem_js_1 = require("./actionitem.js");
const crmmatches_js_1 = require("./crmmatches.js");
const fathomuser_js_1 = require("./fathomuser.js");
const invitee_js_1 = require("./invitee.js");
const meetingsummary_js_1 = require("./meetingsummary.js");
const transcriptitem_js_1 = require("./transcriptitem.js");
exports.CalendarInviteesDomainsType = {
    OnlyInternal: "only_internal",
    OneOrMoreExternal: "one_or_more_external",
};
/** @internal */
exports.CalendarInviteesDomainsType$inboundSchema = z.nativeEnum(exports.CalendarInviteesDomainsType);
/** @internal */
exports.CalendarInviteesDomainsType$outboundSchema = exports.CalendarInviteesDomainsType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalendarInviteesDomainsType$;
(function (CalendarInviteesDomainsType$) {
    /** @deprecated use `CalendarInviteesDomainsType$inboundSchema` instead. */
    CalendarInviteesDomainsType$.inboundSchema = exports.CalendarInviteesDomainsType$inboundSchema;
    /** @deprecated use `CalendarInviteesDomainsType$outboundSchema` instead. */
    CalendarInviteesDomainsType$.outboundSchema = exports.CalendarInviteesDomainsType$outboundSchema;
})(CalendarInviteesDomainsType$ || (exports.CalendarInviteesDomainsType$ = CalendarInviteesDomainsType$ = {}));
/** @internal */
exports.Meeting$inboundSchema = z.object({
    title: z.string(),
    meeting_title: z.nullable(z.string()),
    recording_id: z.number().int(),
    url: z.string(),
    share_url: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    scheduled_start_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    scheduled_end_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    recording_start_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    recording_end_time: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    calendar_invitees_domains_type: exports.CalendarInviteesDomainsType$inboundSchema,
    transcript_language: z.string(),
    transcript: z.nullable(z.array(transcriptitem_js_1.TranscriptItem$inboundSchema)).optional(),
    default_summary: z.nullable(meetingsummary_js_1.MeetingSummary$inboundSchema).optional(),
    action_items: z.nullable(z.array(actionitem_js_1.ActionItem$inboundSchema)).optional(),
    calendar_invitees: z.array(invitee_js_1.Invitee$inboundSchema),
    recorded_by: fathomuser_js_1.FathomUser$inboundSchema,
    crm_matches: z.nullable(crmmatches_js_1.CRMMatches$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "meeting_title": "meetingTitle",
        "recording_id": "recordingId",
        "share_url": "shareUrl",
        "created_at": "createdAt",
        "scheduled_start_time": "scheduledStartTime",
        "scheduled_end_time": "scheduledEndTime",
        "recording_start_time": "recordingStartTime",
        "recording_end_time": "recordingEndTime",
        "calendar_invitees_domains_type": "calendarInviteesDomainsType",
        "transcript_language": "transcriptLanguage",
        "default_summary": "defaultSummary",
        "action_items": "actionItems",
        "calendar_invitees": "calendarInvitees",
        "recorded_by": "recordedBy",
        "crm_matches": "crmMatches",
    });
});
/** @internal */
exports.Meeting$outboundSchema = z.object({
    title: z.string(),
    meetingTitle: z.nullable(z.string()),
    recordingId: z.number().int(),
    url: z.string(),
    shareUrl: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    scheduledStartTime: z.date().transform(v => v.toISOString()),
    scheduledEndTime: z.date().transform(v => v.toISOString()),
    recordingStartTime: z.date().transform(v => v.toISOString()),
    recordingEndTime: z.date().transform(v => v.toISOString()),
    calendarInviteesDomainsType: exports.CalendarInviteesDomainsType$outboundSchema,
    transcriptLanguage: z.string(),
    transcript: z.nullable(z.array(transcriptitem_js_1.TranscriptItem$outboundSchema)).optional(),
    defaultSummary: z.nullable(meetingsummary_js_1.MeetingSummary$outboundSchema).optional(),
    actionItems: z.nullable(z.array(actionitem_js_1.ActionItem$outboundSchema)).optional(),
    calendarInvitees: z.array(invitee_js_1.Invitee$outboundSchema),
    recordedBy: fathomuser_js_1.FathomUser$outboundSchema,
    crmMatches: z.nullable(crmmatches_js_1.CRMMatches$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        meetingTitle: "meeting_title",
        recordingId: "recording_id",
        shareUrl: "share_url",
        createdAt: "created_at",
        scheduledStartTime: "scheduled_start_time",
        scheduledEndTime: "scheduled_end_time",
        recordingStartTime: "recording_start_time",
        recordingEndTime: "recording_end_time",
        calendarInviteesDomainsType: "calendar_invitees_domains_type",
        transcriptLanguage: "transcript_language",
        defaultSummary: "default_summary",
        actionItems: "action_items",
        calendarInvitees: "calendar_invitees",
        recordedBy: "recorded_by",
        crmMatches: "crm_matches",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Meeting$;
(function (Meeting$) {
    /** @deprecated use `Meeting$inboundSchema` instead. */
    Meeting$.inboundSchema = exports.Meeting$inboundSchema;
    /** @deprecated use `Meeting$outboundSchema` instead. */
    Meeting$.outboundSchema = exports.Meeting$outboundSchema;
})(Meeting$ || (exports.Meeting$ = Meeting$ = {}));
function meetingToJSON(meeting) {
    return JSON.stringify(exports.Meeting$outboundSchema.parse(meeting));
}
function meetingFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Meeting$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Meeting' from JSON`);
}
