// fathom-typescript/dist/commonjs/sdk/models/shared/teamlistresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamListResponse$ = exports.TeamListResponse$outboundSchema = exports.TeamListResponse$inboundSchema = void 0;
exports.teamListResponseToJSON = teamListResponseToJSON;
exports.teamListResponseFromJSON = teamListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const team_js_1 = require("./team.js");
/** @internal */
exports.TeamListResponse$inboundSchema = z.object({
    limit: z.number().int(),
    next_cursor: z.nullable(z.string()),
    items: z.array(team_js_1.Team$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "next_cursor": "nextCursor",
    });
});
/** @internal */
exports.TeamListResponse$outboundSchema = z.object({
    limit: z.number().int(),
    nextCursor: z.nullable(z.string()),
    items: z.array(team_js_1.Team$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        nextCursor: "next_cursor",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TeamListResponse$;
(function (TeamListResponse$) {
    /** @deprecated use `TeamListResponse$inboundSchema` instead. */
    TeamListResponse$.inboundSchema = exports.TeamListResponse$inboundSchema;
    /** @deprecated use `TeamListResponse$outboundSchema` instead. */
    TeamListResponse$.outboundSchema = exports.TeamListResponse$outboundSchema;
})(TeamListResponse$ || (exports.TeamListResponse$ = TeamListResponse$ = {}));
function teamListResponseToJSON(teamListResponse) {
    return JSON.stringify(exports.TeamListResponse$outboundSchema.parse(teamListResponse));
}
function teamListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TeamListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TeamListResponse' from JSON`);
}
