// fathom-typescript/dist/commonjs/sdk/models/shared/transcriptitemspeaker.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TranscriptItemSpeaker$ = exports.TranscriptItemSpeaker$outboundSchema = exports.TranscriptItemSpeaker$inboundSchema = void 0;
exports.transcriptItemSpeakerToJSON = transcriptItemSpeakerToJSON;
exports.transcriptItemSpeakerFromJSON = transcriptItemSpeakerFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
/** @internal */
exports.TranscriptItemSpeaker$inboundSchema = z.object({
    display_name: z.string(),
    matched_calendar_invitee_email: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "display_name": "displayName",
        "matched_calendar_invitee_email": "matchedCalendarInviteeEmail",
    });
});
/** @internal */
exports.TranscriptItemSpeaker$outboundSchema = z.object({
    displayName: z.string(),
    matchedCalendarInviteeEmail: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        displayName: "display_name",
        matchedCalendarInviteeEmail: "matched_calendar_invitee_email",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TranscriptItemSpeaker$;
(function (TranscriptItemSpeaker$) {
    /** @deprecated use `TranscriptItemSpeaker$inboundSchema` instead. */
    TranscriptItemSpeaker$.inboundSchema = exports.TranscriptItemSpeaker$inboundSchema;
    /** @deprecated use `TranscriptItemSpeaker$outboundSchema` instead. */
    TranscriptItemSpeaker$.outboundSchema = exports.TranscriptItemSpeaker$outboundSchema;
})(TranscriptItemSpeaker$ || (exports.TranscriptItemSpeaker$ = TranscriptItemSpeaker$ = {}));
function transcriptItemSpeakerToJSON(transcriptItemSpeaker) {
    return JSON.stringify(exports.TranscriptItemSpeaker$outboundSchema.parse(transcriptItemSpeaker));
}
function transcriptItemSpeakerFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TranscriptItemSpeaker$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TranscriptItemSpeaker' from JSON`);
}
