// fathom-typescript/dist/commonjs/sdk/models/shared/webhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook$ = exports.Webhook$outboundSchema = exports.Webhook$inboundSchema = exports.TriggeredFor$ = exports.TriggeredFor$outboundSchema = exports.TriggeredFor$inboundSchema = exports.TriggeredFor = void 0;
exports.webhookToJSON = webhookToJSON;
exports.webhookFromJSON = webhookFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
exports.TriggeredFor = {
    MyRecordings: "my_recordings",
    SharedExternalRecordings: "shared_external_recordings",
    MySharedWithTeamRecordings: "my_shared_with_team_recordings",
    SharedTeamRecordings: "shared_team_recordings",
};
/** @internal */
exports.TriggeredFor$inboundSchema = z.nativeEnum(exports.TriggeredFor);
/** @internal */
exports.TriggeredFor$outboundSchema = exports.TriggeredFor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TriggeredFor$;
(function (TriggeredFor$) {
    /** @deprecated use `TriggeredFor$inboundSchema` instead. */
    TriggeredFor$.inboundSchema = exports.TriggeredFor$inboundSchema;
    /** @deprecated use `TriggeredFor$outboundSchema` instead. */
    TriggeredFor$.outboundSchema = exports.TriggeredFor$outboundSchema;
})(TriggeredFor$ || (exports.TriggeredFor$ = TriggeredFor$ = {}));
/** @internal */
exports.Webhook$inboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    secret: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    include_transcript: z.boolean(),
    include_crm_matches: z.boolean(),
    include_summary: z.boolean(),
    include_action_items: z.boolean(),
    triggered_for: z.array(exports.TriggeredFor$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "include_transcript": "includeTranscript",
        "include_crm_matches": "includeCrmMatches",
        "include_summary": "includeSummary",
        "include_action_items": "includeActionItems",
        "triggered_for": "triggeredFor",
    });
});
/** @internal */
exports.Webhook$outboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    secret: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    includeTranscript: z.boolean(),
    includeCrmMatches: z.boolean(),
    includeSummary: z.boolean(),
    includeActionItems: z.boolean(),
    triggeredFor: z.array(exports.TriggeredFor$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        includeTranscript: "include_transcript",
        includeCrmMatches: "include_crm_matches",
        includeSummary: "include_summary",
        includeActionItems: "include_action_items",
        triggeredFor: "triggered_for",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Webhook$;
(function (Webhook$) {
    /** @deprecated use `Webhook$inboundSchema` instead. */
    Webhook$.inboundSchema = exports.Webhook$inboundSchema;
    /** @deprecated use `Webhook$outboundSchema` instead. */
    Webhook$.outboundSchema = exports.Webhook$outboundSchema;
})(Webhook$ || (exports.Webhook$ = Webhook$ = {}));
function webhookToJSON(webhook) {
    return JSON.stringify(exports.Webhook$outboundSchema.parse(webhook));
}
function webhookFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Webhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Webhook' from JSON`);
}
