// fathom-typescript/dist/esm/sdk/models/operations/createwebhook.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
export const TriggeredFor = {
    MyRecordings: "my_recordings",
    SharedExternalRecordings: "shared_external_recordings",
    MySharedWithTeamRecordings: "my_shared_with_team_recordings",
    SharedTeamRecordings: "shared_team_recordings",
};
/** @internal */
export const TriggeredFor$inboundSchema = z.nativeEnum(TriggeredFor);
/** @internal */
export const TriggeredFor$outboundSchema = TriggeredFor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var TriggeredFor$;
(function (TriggeredFor$) {
    /** @deprecated use `TriggeredFor$inboundSchema` instead. */
    TriggeredFor$.inboundSchema = TriggeredFor$inboundSchema;
    /** @deprecated use `TriggeredFor$outboundSchema` instead. */
    TriggeredFor$.outboundSchema = TriggeredFor$outboundSchema;
})(TriggeredFor$ || (TriggeredFor$ = {}));
/** @internal */
export const CreateWebhookRequest$inboundSchema = z.object({
    destination_url: z.string(),
    include_action_items: z.boolean().default(false),
    include_crm_matches: z.boolean().default(false),
    include_summary: z.boolean().default(false),
    include_transcript: z.boolean().default(false),
    triggered_for: z.array(TriggeredFor$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "destination_url": "destinationUrl",
        "include_action_items": "includeActionItems",
        "include_crm_matches": "includeCrmMatches",
        "include_summary": "includeSummary",
        "include_transcript": "includeTranscript",
        "triggered_for": "triggeredFor",
    });
});
/** @internal */
export const CreateWebhookRequest$outboundSchema = z.object({
    destinationUrl: z.string(),
    includeActionItems: z.boolean().default(false),
    includeCrmMatches: z.boolean().default(false),
    includeSummary: z.boolean().default(false),
    includeTranscript: z.boolean().default(false),
    triggeredFor: z.array(TriggeredFor$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        destinationUrl: "destination_url",
        includeActionItems: "include_action_items",
        includeCrmMatches: "include_crm_matches",
        includeSummary: "include_summary",
        includeTranscript: "include_transcript",
        triggeredFor: "triggered_for",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CreateWebhookRequest$;
(function (CreateWebhookRequest$) {
    /** @deprecated use `CreateWebhookRequest$inboundSchema` instead. */
    CreateWebhookRequest$.inboundSchema = CreateWebhookRequest$inboundSchema;
    /** @deprecated use `CreateWebhookRequest$outboundSchema` instead. */
    CreateWebhookRequest$.outboundSchema = CreateWebhookRequest$outboundSchema;
})(CreateWebhookRequest$ || (CreateWebhookRequest$ = {}));
export function createWebhookRequestToJSON(createWebhookRequest) {
    return JSON.stringify(CreateWebhookRequest$outboundSchema.parse(createWebhookRequest));
}
export function createWebhookRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateWebhookRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateWebhookRequest' from JSON`);
}
