// fathom-typescript/dist/esm/sdk/models/operations/getrecordingsummary.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import * as shared from "../shared/index.js";
/** @internal */
export const GetRecordingSummaryRequest$inboundSchema = z.object({
    recording_id: z.number().int(),
    destination_url: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "recording_id": "recordingId",
        "destination_url": "destinationUrl",
    });
});
/** @internal */
export const GetRecordingSummaryRequest$outboundSchema = z.object({
    recordingId: z.number().int(),
    destinationUrl: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        recordingId: "recording_id",
        destinationUrl: "destination_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetRecordingSummaryRequest$;
(function (GetRecordingSummaryRequest$) {
    /** @deprecated use `GetRecordingSummaryRequest$inboundSchema` instead. */
    GetRecordingSummaryRequest$.inboundSchema = GetRecordingSummaryRequest$inboundSchema;
    /** @deprecated use `GetRecordingSummaryRequest$outboundSchema` instead. */
    GetRecordingSummaryRequest$.outboundSchema = GetRecordingSummaryRequest$outboundSchema;
})(GetRecordingSummaryRequest$ || (GetRecordingSummaryRequest$ = {}));
export function getRecordingSummaryRequestToJSON(getRecordingSummaryRequest) {
    return JSON.stringify(GetRecordingSummaryRequest$outboundSchema.parse(getRecordingSummaryRequest));
}
export function getRecordingSummaryRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRecordingSummaryRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingSummaryRequest' from JSON`);
}
/** @internal */
export const GetRecordingSummaryResponseBody$inboundSchema = z.object({
    summary: z.nullable(shared.MeetingSummary$inboundSchema),
});
/** @internal */
export const GetRecordingSummaryResponseBody$outboundSchema = z.object({
    summary: z.nullable(shared.MeetingSummary$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetRecordingSummaryResponseBody$;
(function (GetRecordingSummaryResponseBody$) {
    /** @deprecated use `GetRecordingSummaryResponseBody$inboundSchema` instead. */
    GetRecordingSummaryResponseBody$.inboundSchema = GetRecordingSummaryResponseBody$inboundSchema;
    /** @deprecated use `GetRecordingSummaryResponseBody$outboundSchema` instead. */
    GetRecordingSummaryResponseBody$.outboundSchema = GetRecordingSummaryResponseBody$outboundSchema;
})(GetRecordingSummaryResponseBody$ || (GetRecordingSummaryResponseBody$ = {}));
export function getRecordingSummaryResponseBodyToJSON(getRecordingSummaryResponseBody) {
    return JSON.stringify(GetRecordingSummaryResponseBody$outboundSchema.parse(getRecordingSummaryResponseBody));
}
export function getRecordingSummaryResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRecordingSummaryResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingSummaryResponseBody' from JSON`);
}
/** @internal */
export const GetRecordingSummaryResponse$inboundSchema = z.union([
    z.lazy(() => GetRecordingSummaryResponseBody$inboundSchema),
    shared.CallbackResponse$inboundSchema,
]);
/** @internal */
export const GetRecordingSummaryResponse$outboundSchema = z.union([
    z.lazy(() => GetRecordingSummaryResponseBody$outboundSchema),
    shared.CallbackResponse$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetRecordingSummaryResponse$;
(function (GetRecordingSummaryResponse$) {
    /** @deprecated use `GetRecordingSummaryResponse$inboundSchema` instead. */
    GetRecordingSummaryResponse$.inboundSchema = GetRecordingSummaryResponse$inboundSchema;
    /** @deprecated use `GetRecordingSummaryResponse$outboundSchema` instead. */
    GetRecordingSummaryResponse$.outboundSchema = GetRecordingSummaryResponse$outboundSchema;
})(GetRecordingSummaryResponse$ || (GetRecordingSummaryResponse$ = {}));
export function getRecordingSummaryResponseToJSON(getRecordingSummaryResponse) {
    return JSON.stringify(GetRecordingSummaryResponse$outboundSchema.parse(getRecordingSummaryResponse));
}
export function getRecordingSummaryResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRecordingSummaryResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingSummaryResponse' from JSON`);
}
