// fathom-typescript/dist/esm/sdk/models/operations/getrecordingtranscript.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import * as shared from "../shared/index.js";
/** @internal */
export const GetRecordingTranscriptRequest$inboundSchema = z.object({
    recording_id: z.number().int(),
    destination_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "recording_id": "recordingId",
        "destination_url": "destinationUrl",
    });
});
/** @internal */
export const GetRecordingTranscriptRequest$outboundSchema = z.object({
    recordingId: z.number().int(),
    destinationUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        recordingId: "recording_id",
        destinationUrl: "destination_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetRecordingTranscriptRequest$;
(function (GetRecordingTranscriptRequest$) {
    /** @deprecated use `GetRecordingTranscriptRequest$inboundSchema` instead. */
    GetRecordingTranscriptRequest$.inboundSchema = GetRecordingTranscriptRequest$inboundSchema;
    /** @deprecated use `GetRecordingTranscriptRequest$outboundSchema` instead. */
    GetRecordingTranscriptRequest$.outboundSchema = GetRecordingTranscriptRequest$outboundSchema;
})(GetRecordingTranscriptRequest$ || (GetRecordingTranscriptRequest$ = {}));
export function getRecordingTranscriptRequestToJSON(getRecordingTranscriptRequest) {
    return JSON.stringify(GetRecordingTranscriptRequest$outboundSchema.parse(getRecordingTranscriptRequest));
}
export function getRecordingTranscriptRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRecordingTranscriptRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingTranscriptRequest' from JSON`);
}
/** @internal */
export const GetRecordingTranscriptResponseBody$inboundSchema = z.object({
    transcript: z.array(shared.TranscriptItem$inboundSchema),
});
/** @internal */
export const GetRecordingTranscriptResponseBody$outboundSchema = z.object({
    transcript: z.array(shared.TranscriptItem$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetRecordingTranscriptResponseBody$;
(function (GetRecordingTranscriptResponseBody$) {
    /** @deprecated use `GetRecordingTranscriptResponseBody$inboundSchema` instead. */
    GetRecordingTranscriptResponseBody$.inboundSchema = GetRecordingTranscriptResponseBody$inboundSchema;
    /** @deprecated use `GetRecordingTranscriptResponseBody$outboundSchema` instead. */
    GetRecordingTranscriptResponseBody$.outboundSchema = GetRecordingTranscriptResponseBody$outboundSchema;
})(GetRecordingTranscriptResponseBody$ || (GetRecordingTranscriptResponseBody$ = {}));
export function getRecordingTranscriptResponseBodyToJSON(getRecordingTranscriptResponseBody) {
    return JSON.stringify(GetRecordingTranscriptResponseBody$outboundSchema.parse(getRecordingTranscriptResponseBody));
}
export function getRecordingTranscriptResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRecordingTranscriptResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingTranscriptResponseBody' from JSON`);
}
/** @internal */
export const GetRecordingTranscriptResponse$inboundSchema = z.union([
    z.lazy(() => GetRecordingTranscriptResponseBody$inboundSchema),
    shared.CallbackResponse$inboundSchema,
]);
/** @internal */
export const GetRecordingTranscriptResponse$outboundSchema = z.union([
    z.lazy(() => GetRecordingTranscriptResponseBody$outboundSchema),
    shared.CallbackResponse$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetRecordingTranscriptResponse$;
(function (GetRecordingTranscriptResponse$) {
    /** @deprecated use `GetRecordingTranscriptResponse$inboundSchema` instead. */
    GetRecordingTranscriptResponse$.inboundSchema = GetRecordingTranscriptResponse$inboundSchema;
    /** @deprecated use `GetRecordingTranscriptResponse$outboundSchema` instead. */
    GetRecordingTranscriptResponse$.outboundSchema = GetRecordingTranscriptResponse$outboundSchema;
})(GetRecordingTranscriptResponse$ || (GetRecordingTranscriptResponse$ = {}));
export function getRecordingTranscriptResponseToJSON(getRecordingTranscriptResponse) {
    return JSON.stringify(GetRecordingTranscriptResponse$outboundSchema.parse(getRecordingTranscriptResponse));
}
export function getRecordingTranscriptResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRecordingTranscriptResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingTranscriptResponse' from JSON`);
}
