// fathom-typescript/dist/esm/sdk/models/operations/listteams.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import * as shared from "../shared/index.js";
/** @internal */
export const ListTeamsRequest$inboundSchema = z.object({
    cursor: z.string().optional(),
});
/** @internal */
export const ListTeamsRequest$outboundSchema = z.object({
    cursor: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListTeamsRequest$;
(function (ListTeamsRequest$) {
    /** @deprecated use `ListTeamsRequest$inboundSchema` instead. */
    ListTeamsRequest$.inboundSchema = ListTeamsRequest$inboundSchema;
    /** @deprecated use `ListTeamsRequest$outboundSchema` instead. */
    ListTeamsRequest$.outboundSchema = ListTeamsRequest$outboundSchema;
})(ListTeamsRequest$ || (ListTeamsRequest$ = {}));
export function listTeamsRequestToJSON(listTeamsRequest) {
    return JSON.stringify(ListTeamsRequest$outboundSchema.parse(listTeamsRequest));
}
export function listTeamsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListTeamsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamsRequest' from JSON`);
}
/** @internal */
export const ListTeamsResponse$inboundSchema = z.object({
    Result: shared.TeamListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
/** @internal */
export const ListTeamsResponse$outboundSchema = z.object({
    result: shared.TeamListResponse$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListTeamsResponse$;
(function (ListTeamsResponse$) {
    /** @deprecated use `ListTeamsResponse$inboundSchema` instead. */
    ListTeamsResponse$.inboundSchema = ListTeamsResponse$inboundSchema;
    /** @deprecated use `ListTeamsResponse$outboundSchema` instead. */
    ListTeamsResponse$.outboundSchema = ListTeamsResponse$outboundSchema;
})(ListTeamsResponse$ || (ListTeamsResponse$ = {}));
export function listTeamsResponseToJSON(listTeamsResponse) {
    return JSON.stringify(ListTeamsResponse$outboundSchema.parse(listTeamsResponse));
}
export function listTeamsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListTeamsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamsResponse' from JSON`);
}
