// fathom-typescript/dist/esm/sdk/models/operations/listmeetings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import * as shared from "../shared/index.js";
/**
 * Filter by whether calendar invitee list includes external email domains.
 */
export const CalendarInviteesDomainsType = {
    All: "all",
    OnlyInternal: "only_internal",
    OneOrMoreExternal: "one_or_more_external",
};
/**
 * Filter by meeting type.
 */
export const MeetingType = {
    All: "all",
    Internal: "internal",
    External: "external",
};
/** @internal */
export const CalendarInviteesDomainsType$inboundSchema = z.nativeEnum(CalendarInviteesDomainsType);
/** @internal */
export const CalendarInviteesDomainsType$outboundSchema = CalendarInviteesDomainsType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CalendarInviteesDomainsType$;
(function (CalendarInviteesDomainsType$) {
    /** @deprecated use `CalendarInviteesDomainsType$inboundSchema` instead. */
    CalendarInviteesDomainsType$.inboundSchema = CalendarInviteesDomainsType$inboundSchema;
    /** @deprecated use `CalendarInviteesDomainsType$outboundSchema` instead. */
    CalendarInviteesDomainsType$.outboundSchema = CalendarInviteesDomainsType$outboundSchema;
})(CalendarInviteesDomainsType$ || (CalendarInviteesDomainsType$ = {}));
/** @internal */
export const MeetingType$inboundSchema = z
    .nativeEnum(MeetingType);
/** @internal */
export const MeetingType$outboundSchema = MeetingType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var MeetingType$;
(function (MeetingType$) {
    /** @deprecated use `MeetingType$inboundSchema` instead. */
    MeetingType$.inboundSchema = MeetingType$inboundSchema;
    /** @deprecated use `MeetingType$outboundSchema` instead. */
    MeetingType$.outboundSchema = MeetingType$outboundSchema;
})(MeetingType$ || (MeetingType$ = {}));
/** @internal */
export const ListMeetingsRequest$inboundSchema = z.object({
    "calendar_invitees[]": z.array(z.string()).optional(),
    "calendar_invitees_domains[]": z.array(z.string()).optional(),
    calendar_invitees_domains_type: CalendarInviteesDomainsType$inboundSchema
        .default("all"),
    created_after: z.string().optional(),
    created_before: z.string().optional(),
    cursor: z.string().optional(),
    include_action_items: z.boolean().default(false),
    include_crm_matches: z.boolean().default(false),
    include_summary: z.boolean().default(false),
    include_transcript: z.boolean().default(false),
    meeting_type: MeetingType$inboundSchema.optional(),
    "recorded_by[]": z.array(z.string()).optional(),
    "teams[]": z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "calendar_invitees[]": "calendarInvitees",
        "calendar_invitees_domains[]": "calendarInviteesDomains",
        "calendar_invitees_domains_type": "calendarInviteesDomainsType",
        "created_after": "createdAfter",
        "created_before": "createdBefore",
        "include_action_items": "includeActionItems",
        "include_crm_matches": "includeCrmMatches",
        "include_summary": "includeSummary",
        "include_transcript": "includeTranscript",
        "meeting_type": "meetingType",
        "recorded_by[]": "recordedBy",
        "teams[]": "teams",
    });
});
/** @internal */
export const ListMeetingsRequest$outboundSchema = z.object({
    calendarInvitees: z.array(z.string()).optional(),
    calendarInviteesDomains: z.array(z.string()).optional(),
    calendarInviteesDomainsType: CalendarInviteesDomainsType$outboundSchema
        .default("all"),
    createdAfter: z.string().optional(),
    createdBefore: z.string().optional(),
    cursor: z.string().optional(),
    includeActionItems: z.boolean().default(false),
    includeCrmMatches: z.boolean().default(false),
    includeSummary: z.boolean().default(false),
    includeTranscript: z.boolean().default(false),
    meetingType: MeetingType$outboundSchema.optional(),
    recordedBy: z.array(z.string()).optional(),
    teams: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        calendarInvitees: "calendar_invitees[]",
        calendarInviteesDomains: "calendar_invitees_domains[]",
        calendarInviteesDomainsType: "calendar_invitees_domains_type",
        createdAfter: "created_after",
        createdBefore: "created_before",
        includeActionItems: "include_action_items",
        includeCrmMatches: "include_crm_matches",
        includeSummary: "include_summary",
        includeTranscript: "include_transcript",
        meetingType: "meeting_type",
        recordedBy: "recorded_by[]",
        teams: "teams[]",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListMeetingsRequest$;
(function (ListMeetingsRequest$) {
    /** @deprecated use `ListMeetingsRequest$inboundSchema` instead. */
    ListMeetingsRequest$.inboundSchema = ListMeetingsRequest$inboundSchema;
    /** @deprecated use `ListMeetingsRequest$outboundSchema` instead. */
    ListMeetingsRequest$.outboundSchema = ListMeetingsRequest$outboundSchema;
})(ListMeetingsRequest$ || (ListMeetingsRequest$ = {}));
export function listMeetingsRequestToJSON(listMeetingsRequest) {
    return JSON.stringify(ListMeetingsRequest$outboundSchema.parse(listMeetingsRequest));
}
export function listMeetingsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListMeetingsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListMeetingsRequest' from JSON`);
}
/** @internal */
export const ListMeetingsResponse$inboundSchema = z.object({
    Result: shared.MeetingListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
/** @internal */
export const ListMeetingsResponse$outboundSchema = z.object({
    result: shared.MeetingListResponse$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListMeetingsResponse$;
(function (ListMeetingsResponse$) {
    /** @deprecated use `ListMeetingsResponse$inboundSchema` instead. */
    ListMeetingsResponse$.inboundSchema = ListMeetingsResponse$inboundSchema;
    /** @deprecated use `ListMeetingsResponse$outboundSchema` instead. */
    ListMeetingsResponse$.outboundSchema = ListMeetingsResponse$outboundSchema;
})(ListMeetingsResponse$ || (ListMeetingsResponse$ = {}));
export function listMeetingsResponseToJSON(listMeetingsResponse) {
    return JSON.stringify(ListMeetingsResponse$outboundSchema.parse(listMeetingsResponse));
}
export function listMeetingsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListMeetingsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListMeetingsResponse' from JSON`);
}
