// fathom-typescript/dist/esm/sdk/models/operations/listteammembers.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import * as shared from "../shared/index.js";
/** @internal */
export const ListTeamMembersRequest$inboundSchema = z.object({
    cursor: z.string().optional(),
    team: z.string().optional(),
});
/** @internal */
export const ListTeamMembersRequest$outboundSchema = z.object({
    cursor: z.string().optional(),
    team: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListTeamMembersRequest$;
(function (ListTeamMembersRequest$) {
    /** @deprecated use `ListTeamMembersRequest$inboundSchema` instead. */
    ListTeamMembersRequest$.inboundSchema = ListTeamMembersRequest$inboundSchema;
    /** @deprecated use `ListTeamMembersRequest$outboundSchema` instead. */
    ListTeamMembersRequest$.outboundSchema = ListTeamMembersRequest$outboundSchema;
})(ListTeamMembersRequest$ || (ListTeamMembersRequest$ = {}));
export function listTeamMembersRequestToJSON(listTeamMembersRequest) {
    return JSON.stringify(ListTeamMembersRequest$outboundSchema.parse(listTeamMembersRequest));
}
export function listTeamMembersRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListTeamMembersRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamMembersRequest' from JSON`);
}
/** @internal */
export const ListTeamMembersResponse$inboundSchema = z.object({
    Result: shared.TeamMemberListResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
/** @internal */
export const ListTeamMembersResponse$outboundSchema = z.object({
    result: shared.TeamMemberListResponse$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ListTeamMembersResponse$;
(function (ListTeamMembersResponse$) {
    /** @deprecated use `ListTeamMembersResponse$inboundSchema` instead. */
    ListTeamMembersResponse$.inboundSchema = ListTeamMembersResponse$inboundSchema;
    /** @deprecated use `ListTeamMembersResponse$outboundSchema` instead. */
    ListTeamMembersResponse$.outboundSchema = ListTeamMembersResponse$outboundSchema;
})(ListTeamMembersResponse$ || (ListTeamMembersResponse$ = {}));
export function listTeamMembersResponseToJSON(listTeamMembersResponse) {
    return JSON.stringify(ListTeamMembersResponse$outboundSchema.parse(listTeamMembersResponse));
}
export function listTeamMembersResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListTeamMembersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamMembersResponse' from JSON`);
}
