// fathom-typescript/dist/esm/funcs/listMeetings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { dlv } from "../lib/dlv.js";
import { encodeFormQuery } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as operations from "../sdk/models/operations/index.js";
import { APIPromise } from "../sdk/types/async.js";
import { createPageIterator, haltIterator, } from "../sdk/types/operations.js";
/**
 * List meetings
 */
export function listMeetings(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => operations.ListMeetingsRequest$outboundSchema.optional().parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [haltIterator(parsed), { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const path = pathToFunc("/meetings")();
    const query = encodeFormQuery({
        "calendar_invitees_domains_type": payload?.calendar_invitees_domains_type,
        "calendar_invitees_domains[]": payload?.["calendar_invitees_domains[]"],
        "calendar_invitees[]": payload?.["calendar_invitees[]"],
        "created_after": payload?.created_after,
        "created_before": payload?.created_before,
        "cursor": payload?.cursor,
        "include_action_items": payload?.include_action_items,
        "include_crm_matches": payload?.include_crm_matches,
        "include_summary": payload?.include_summary,
        "include_transcript": payload?.include_transcript,
        "meeting_type": payload?.meeting_type,
        "recorded_by[]": payload?.["recorded_by[]"],
        "teams[]": payload?.["teams[]"],
    });
    const headers = new Headers(compactMap({
        Accept: "application/json",
    }));
    const securityInput = await extractSecurity(client._options.security);
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "listMeetings",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.security,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        query: query,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [haltIterator(requestRes), { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: [],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [haltIterator(doResult), { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result, raw] = await M.match(M.json(200, operations.ListMeetingsResponse$inboundSchema.optional(), {
        key: "Result",
    }), M.nil([400, 401, 429], operations.ListMeetingsResponse$inboundSchema.optional()))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [haltIterator(result), {
                status: "complete",
                request: req,
                response,
            }];
    }
    const nextFunc = (responseData) => {
        const nextCursor = dlv(responseData, "next_cursor");
        if (typeof nextCursor !== "string") {
            return { next: () => null };
        }
        if (nextCursor.trim() === "") {
            return { next: () => null };
        }
        const nextVal = () => listMeetings(client, {
            ...request,
            cursor: nextCursor,
        }, options);
        return { next: nextVal, "~next": { cursor: nextCursor } };
    };
    const page = { ...result, ...nextFunc(raw) };
    return [{ ...page, ...createPageIterator(page, (v) => !v.ok) }, {
            status: "complete",
            request: req,
            response,
        }];
}
