// fathom-typescript/dist/commonjs/sdk/models/operations/createwebhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateWebhookRequest$ = exports.CreateWebhookRequest$outboundSchema = exports.CreateWebhookRequest$inboundSchema = exports.TriggeredFor$ = exports.TriggeredFor$outboundSchema = exports.TriggeredFor$inboundSchema = exports.TriggeredFor = void 0;
exports.createWebhookRequestToJSON = createWebhookRequestToJSON;
exports.createWebhookRequestFromJSON = createWebhookRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
exports.TriggeredFor = {
    MyRecordings: "my_recordings",
    SharedExternalRecordings: "shared_external_recordings",
    MySharedWithTeamRecordings: "my_shared_with_team_recordings",
    SharedTeamRecordings: "shared_team_recordings",
};
/** @internal */
exports.TriggeredFor$inboundSchema = z.nativeEnum(exports.TriggeredFor);
/** @internal */
exports.TriggeredFor$outboundSchema = exports.TriggeredFor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TriggeredFor$;
(function (TriggeredFor$) {
    /** @deprecated use `TriggeredFor$inboundSchema` instead. */
    TriggeredFor$.inboundSchema = exports.TriggeredFor$inboundSchema;
    /** @deprecated use `TriggeredFor$outboundSchema` instead. */
    TriggeredFor$.outboundSchema = exports.TriggeredFor$outboundSchema;
})(TriggeredFor$ || (exports.TriggeredFor$ = TriggeredFor$ = {}));
/** @internal */
exports.CreateWebhookRequest$inboundSchema = z.object({
    destination_url: z.string(),
    include_action_items: z.boolean().default(false),
    include_crm_matches: z.boolean().default(false),
    include_summary: z.boolean().default(false),
    include_transcript: z.boolean().default(false),
    triggered_for: z.array(exports.TriggeredFor$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "destination_url": "destinationUrl",
        "include_action_items": "includeActionItems",
        "include_crm_matches": "includeCrmMatches",
        "include_summary": "includeSummary",
        "include_transcript": "includeTranscript",
        "triggered_for": "triggeredFor",
    });
});
/** @internal */
exports.CreateWebhookRequest$outboundSchema = z.object({
    destinationUrl: z.string(),
    includeActionItems: z.boolean().default(false),
    includeCrmMatches: z.boolean().default(false),
    includeSummary: z.boolean().default(false),
    includeTranscript: z.boolean().default(false),
    triggeredFor: z.array(exports.TriggeredFor$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        destinationUrl: "destination_url",
        includeActionItems: "include_action_items",
        includeCrmMatches: "include_crm_matches",
        includeSummary: "include_summary",
        includeTranscript: "include_transcript",
        triggeredFor: "triggered_for",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateWebhookRequest$;
(function (CreateWebhookRequest$) {
    /** @deprecated use `CreateWebhookRequest$inboundSchema` instead. */
    CreateWebhookRequest$.inboundSchema = exports.CreateWebhookRequest$inboundSchema;
    /** @deprecated use `CreateWebhookRequest$outboundSchema` instead. */
    CreateWebhookRequest$.outboundSchema = exports.CreateWebhookRequest$outboundSchema;
})(CreateWebhookRequest$ || (exports.CreateWebhookRequest$ = CreateWebhookRequest$ = {}));
function createWebhookRequestToJSON(createWebhookRequest) {
    return JSON.stringify(exports.CreateWebhookRequest$outboundSchema.parse(createWebhookRequest));
}
function createWebhookRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateWebhookRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateWebhookRequest' from JSON`);
}
