// fathom-typescript/dist/commonjs/sdk/models/operations/getrecordingtranscript.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRecordingTranscriptResponse$ = exports.GetRecordingTranscriptResponse$outboundSchema = exports.GetRecordingTranscriptResponse$inboundSchema = exports.GetRecordingTranscriptResponseBody$ = exports.GetRecordingTranscriptResponseBody$outboundSchema = exports.GetRecordingTranscriptResponseBody$inboundSchema = exports.GetRecordingTranscriptRequest$ = exports.GetRecordingTranscriptRequest$outboundSchema = exports.GetRecordingTranscriptRequest$inboundSchema = void 0;
exports.getRecordingTranscriptRequestToJSON = getRecordingTranscriptRequestToJSON;
exports.getRecordingTranscriptRequestFromJSON = getRecordingTranscriptRequestFromJSON;
exports.getRecordingTranscriptResponseBodyToJSON = getRecordingTranscriptResponseBodyToJSON;
exports.getRecordingTranscriptResponseBodyFromJSON = getRecordingTranscriptResponseBodyFromJSON;
exports.getRecordingTranscriptResponseToJSON = getRecordingTranscriptResponseToJSON;
exports.getRecordingTranscriptResponseFromJSON = getRecordingTranscriptResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.GetRecordingTranscriptRequest$inboundSchema = z.object({
    recording_id: z.number().int(),
    destination_url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "recording_id": "recordingId",
        "destination_url": "destinationUrl",
    });
});
/** @internal */
exports.GetRecordingTranscriptRequest$outboundSchema = z.object({
    recordingId: z.number().int(),
    destinationUrl: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        recordingId: "recording_id",
        destinationUrl: "destination_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordingTranscriptRequest$;
(function (GetRecordingTranscriptRequest$) {
    /** @deprecated use `GetRecordingTranscriptRequest$inboundSchema` instead. */
    GetRecordingTranscriptRequest$.inboundSchema = exports.GetRecordingTranscriptRequest$inboundSchema;
    /** @deprecated use `GetRecordingTranscriptRequest$outboundSchema` instead. */
    GetRecordingTranscriptRequest$.outboundSchema = exports.GetRecordingTranscriptRequest$outboundSchema;
})(GetRecordingTranscriptRequest$ || (exports.GetRecordingTranscriptRequest$ = GetRecordingTranscriptRequest$ = {}));
function getRecordingTranscriptRequestToJSON(getRecordingTranscriptRequest) {
    return JSON.stringify(exports.GetRecordingTranscriptRequest$outboundSchema.parse(getRecordingTranscriptRequest));
}
function getRecordingTranscriptRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordingTranscriptRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingTranscriptRequest' from JSON`);
}
/** @internal */
exports.GetRecordingTranscriptResponseBody$inboundSchema = z.object({
    transcript: z.array(shared.TranscriptItem$inboundSchema),
});
/** @internal */
exports.GetRecordingTranscriptResponseBody$outboundSchema = z.object({
    transcript: z.array(shared.TranscriptItem$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordingTranscriptResponseBody$;
(function (GetRecordingTranscriptResponseBody$) {
    /** @deprecated use `GetRecordingTranscriptResponseBody$inboundSchema` instead. */
    GetRecordingTranscriptResponseBody$.inboundSchema = exports.GetRecordingTranscriptResponseBody$inboundSchema;
    /** @deprecated use `GetRecordingTranscriptResponseBody$outboundSchema` instead. */
    GetRecordingTranscriptResponseBody$.outboundSchema = exports.GetRecordingTranscriptResponseBody$outboundSchema;
})(GetRecordingTranscriptResponseBody$ || (exports.GetRecordingTranscriptResponseBody$ = GetRecordingTranscriptResponseBody$ = {}));
function getRecordingTranscriptResponseBodyToJSON(getRecordingTranscriptResponseBody) {
    return JSON.stringify(exports.GetRecordingTranscriptResponseBody$outboundSchema.parse(getRecordingTranscriptResponseBody));
}
function getRecordingTranscriptResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordingTranscriptResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingTranscriptResponseBody' from JSON`);
}
/** @internal */
exports.GetRecordingTranscriptResponse$inboundSchema = z.union([
    z.lazy(() => exports.GetRecordingTranscriptResponseBody$inboundSchema),
    shared.CallbackResponse$inboundSchema,
]);
/** @internal */
exports.GetRecordingTranscriptResponse$outboundSchema = z.union([
    z.lazy(() => exports.GetRecordingTranscriptResponseBody$outboundSchema),
    shared.CallbackResponse$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordingTranscriptResponse$;
(function (GetRecordingTranscriptResponse$) {
    /** @deprecated use `GetRecordingTranscriptResponse$inboundSchema` instead. */
    GetRecordingTranscriptResponse$.inboundSchema = exports.GetRecordingTranscriptResponse$inboundSchema;
    /** @deprecated use `GetRecordingTranscriptResponse$outboundSchema` instead. */
    GetRecordingTranscriptResponse$.outboundSchema = exports.GetRecordingTranscriptResponse$outboundSchema;
})(GetRecordingTranscriptResponse$ || (exports.GetRecordingTranscriptResponse$ = GetRecordingTranscriptResponse$ = {}));
function getRecordingTranscriptResponseToJSON(getRecordingTranscriptResponse) {
    return JSON.stringify(exports.GetRecordingTranscriptResponse$outboundSchema.parse(getRecordingTranscriptResponse));
}
function getRecordingTranscriptResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordingTranscriptResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingTranscriptResponse' from JSON`);
}
