// fathom-typescript/dist/commonjs/sdk/models/operations/getrecordingsummary.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRecordingSummaryResponse$ = exports.GetRecordingSummaryResponse$outboundSchema = exports.GetRecordingSummaryResponse$inboundSchema = exports.GetRecordingSummaryResponseBody$ = exports.GetRecordingSummaryResponseBody$outboundSchema = exports.GetRecordingSummaryResponseBody$inboundSchema = exports.GetRecordingSummaryRequest$ = exports.GetRecordingSummaryRequest$outboundSchema = exports.GetRecordingSummaryRequest$inboundSchema = void 0;
exports.getRecordingSummaryRequestToJSON = getRecordingSummaryRequestToJSON;
exports.getRecordingSummaryRequestFromJSON = getRecordingSummaryRequestFromJSON;
exports.getRecordingSummaryResponseBodyToJSON = getRecordingSummaryResponseBodyToJSON;
exports.getRecordingSummaryResponseBodyFromJSON = getRecordingSummaryResponseBodyFromJSON;
exports.getRecordingSummaryResponseToJSON = getRecordingSummaryResponseToJSON;
exports.getRecordingSummaryResponseFromJSON = getRecordingSummaryResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.GetRecordingSummaryRequest$inboundSchema = z.object({
    recording_id: z.number().int(),
    destination_url: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "recording_id": "recordingId",
        "destination_url": "destinationUrl",
    });
});
/** @internal */
exports.GetRecordingSummaryRequest$outboundSchema = z.object({
    recordingId: z.number().int(),
    destinationUrl: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        recordingId: "recording_id",
        destinationUrl: "destination_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordingSummaryRequest$;
(function (GetRecordingSummaryRequest$) {
    /** @deprecated use `GetRecordingSummaryRequest$inboundSchema` instead. */
    GetRecordingSummaryRequest$.inboundSchema = exports.GetRecordingSummaryRequest$inboundSchema;
    /** @deprecated use `GetRecordingSummaryRequest$outboundSchema` instead. */
    GetRecordingSummaryRequest$.outboundSchema = exports.GetRecordingSummaryRequest$outboundSchema;
})(GetRecordingSummaryRequest$ || (exports.GetRecordingSummaryRequest$ = GetRecordingSummaryRequest$ = {}));
function getRecordingSummaryRequestToJSON(getRecordingSummaryRequest) {
    return JSON.stringify(exports.GetRecordingSummaryRequest$outboundSchema.parse(getRecordingSummaryRequest));
}
function getRecordingSummaryRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordingSummaryRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingSummaryRequest' from JSON`);
}
/** @internal */
exports.GetRecordingSummaryResponseBody$inboundSchema = z.object({
    summary: z.nullable(shared.MeetingSummary$inboundSchema),
});
/** @internal */
exports.GetRecordingSummaryResponseBody$outboundSchema = z.object({
    summary: z.nullable(shared.MeetingSummary$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordingSummaryResponseBody$;
(function (GetRecordingSummaryResponseBody$) {
    /** @deprecated use `GetRecordingSummaryResponseBody$inboundSchema` instead. */
    GetRecordingSummaryResponseBody$.inboundSchema = exports.GetRecordingSummaryResponseBody$inboundSchema;
    /** @deprecated use `GetRecordingSummaryResponseBody$outboundSchema` instead. */
    GetRecordingSummaryResponseBody$.outboundSchema = exports.GetRecordingSummaryResponseBody$outboundSchema;
})(GetRecordingSummaryResponseBody$ || (exports.GetRecordingSummaryResponseBody$ = GetRecordingSummaryResponseBody$ = {}));
function getRecordingSummaryResponseBodyToJSON(getRecordingSummaryResponseBody) {
    return JSON.stringify(exports.GetRecordingSummaryResponseBody$outboundSchema.parse(getRecordingSummaryResponseBody));
}
function getRecordingSummaryResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordingSummaryResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingSummaryResponseBody' from JSON`);
}
/** @internal */
exports.GetRecordingSummaryResponse$inboundSchema = z.union([
    z.lazy(() => exports.GetRecordingSummaryResponseBody$inboundSchema),
    shared.CallbackResponse$inboundSchema,
]);
/** @internal */
exports.GetRecordingSummaryResponse$outboundSchema = z.union([
    z.lazy(() => exports.GetRecordingSummaryResponseBody$outboundSchema),
    shared.CallbackResponse$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRecordingSummaryResponse$;
(function (GetRecordingSummaryResponse$) {
    /** @deprecated use `GetRecordingSummaryResponse$inboundSchema` instead. */
    GetRecordingSummaryResponse$.inboundSchema = exports.GetRecordingSummaryResponse$inboundSchema;
    /** @deprecated use `GetRecordingSummaryResponse$outboundSchema` instead. */
    GetRecordingSummaryResponse$.outboundSchema = exports.GetRecordingSummaryResponse$outboundSchema;
})(GetRecordingSummaryResponse$ || (exports.GetRecordingSummaryResponse$ = GetRecordingSummaryResponse$ = {}));
function getRecordingSummaryResponseToJSON(getRecordingSummaryResponse) {
    return JSON.stringify(exports.GetRecordingSummaryResponse$outboundSchema.parse(getRecordingSummaryResponse));
}
function getRecordingSummaryResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRecordingSummaryResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRecordingSummaryResponse' from JSON`);
}
