// fathom-typescript/dist/commonjs/sdk/models/operations/listteams.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTeamsResponse$ = exports.ListTeamsResponse$outboundSchema = exports.ListTeamsResponse$inboundSchema = exports.ListTeamsRequest$ = exports.ListTeamsRequest$outboundSchema = exports.ListTeamsRequest$inboundSchema = void 0;
exports.listTeamsRequestToJSON = listTeamsRequestToJSON;
exports.listTeamsRequestFromJSON = listTeamsRequestFromJSON;
exports.listTeamsResponseToJSON = listTeamsResponseToJSON;
exports.listTeamsResponseFromJSON = listTeamsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.ListTeamsRequest$inboundSchema = z.object({
    cursor: z.string().optional(),
});
/** @internal */
exports.ListTeamsRequest$outboundSchema = z.object({
    cursor: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListTeamsRequest$;
(function (ListTeamsRequest$) {
    /** @deprecated use `ListTeamsRequest$inboundSchema` instead. */
    ListTeamsRequest$.inboundSchema = exports.ListTeamsRequest$inboundSchema;
    /** @deprecated use `ListTeamsRequest$outboundSchema` instead. */
    ListTeamsRequest$.outboundSchema = exports.ListTeamsRequest$outboundSchema;
})(ListTeamsRequest$ || (exports.ListTeamsRequest$ = ListTeamsRequest$ = {}));
function listTeamsRequestToJSON(listTeamsRequest) {
    return JSON.stringify(exports.ListTeamsRequest$outboundSchema.parse(listTeamsRequest));
}
function listTeamsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListTeamsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamsRequest' from JSON`);
}
/** @internal */
exports.ListTeamsResponse$inboundSchema = z.object({
    Result: shared.TeamListResponse$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ListTeamsResponse$outboundSchema = z.object({
    result: shared.TeamListResponse$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListTeamsResponse$;
(function (ListTeamsResponse$) {
    /** @deprecated use `ListTeamsResponse$inboundSchema` instead. */
    ListTeamsResponse$.inboundSchema = exports.ListTeamsResponse$inboundSchema;
    /** @deprecated use `ListTeamsResponse$outboundSchema` instead. */
    ListTeamsResponse$.outboundSchema = exports.ListTeamsResponse$outboundSchema;
})(ListTeamsResponse$ || (exports.ListTeamsResponse$ = ListTeamsResponse$ = {}));
function listTeamsResponseToJSON(listTeamsResponse) {
    return JSON.stringify(exports.ListTeamsResponse$outboundSchema.parse(listTeamsResponse));
}
function listTeamsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListTeamsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamsResponse' from JSON`);
}
