// fathom-typescript/dist/commonjs/sdk/models/operations/listmeetings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMeetingsResponse$ = exports.ListMeetingsResponse$outboundSchema = exports.ListMeetingsResponse$inboundSchema = exports.ListMeetingsRequest$ = exports.ListMeetingsRequest$outboundSchema = exports.ListMeetingsRequest$inboundSchema = exports.MeetingType$ = exports.MeetingType$outboundSchema = exports.MeetingType$inboundSchema = exports.CalendarInviteesDomainsType$ = exports.CalendarInviteesDomainsType$outboundSchema = exports.CalendarInviteesDomainsType$inboundSchema = exports.MeetingType = exports.CalendarInviteesDomainsType = void 0;
exports.listMeetingsRequestToJSON = listMeetingsRequestToJSON;
exports.listMeetingsRequestFromJSON = listMeetingsRequestFromJSON;
exports.listMeetingsResponseToJSON = listMeetingsResponseToJSON;
exports.listMeetingsResponseFromJSON = listMeetingsResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/**
 * Filter by whether calendar invitee list includes external email domains.
 */
exports.CalendarInviteesDomainsType = {
    All: "all",
    OnlyInternal: "only_internal",
    OneOrMoreExternal: "one_or_more_external",
};
/**
 * Filter by meeting type.
 */
exports.MeetingType = {
    All: "all",
    Internal: "internal",
    External: "external",
};
/** @internal */
exports.CalendarInviteesDomainsType$inboundSchema = z.nativeEnum(exports.CalendarInviteesDomainsType);
/** @internal */
exports.CalendarInviteesDomainsType$outboundSchema = exports.CalendarInviteesDomainsType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalendarInviteesDomainsType$;
(function (CalendarInviteesDomainsType$) {
    /** @deprecated use `CalendarInviteesDomainsType$inboundSchema` instead. */
    CalendarInviteesDomainsType$.inboundSchema = exports.CalendarInviteesDomainsType$inboundSchema;
    /** @deprecated use `CalendarInviteesDomainsType$outboundSchema` instead. */
    CalendarInviteesDomainsType$.outboundSchema = exports.CalendarInviteesDomainsType$outboundSchema;
})(CalendarInviteesDomainsType$ || (exports.CalendarInviteesDomainsType$ = CalendarInviteesDomainsType$ = {}));
/** @internal */
exports.MeetingType$inboundSchema = z
    .nativeEnum(exports.MeetingType);
/** @internal */
exports.MeetingType$outboundSchema = exports.MeetingType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MeetingType$;
(function (MeetingType$) {
    /** @deprecated use `MeetingType$inboundSchema` instead. */
    MeetingType$.inboundSchema = exports.MeetingType$inboundSchema;
    /** @deprecated use `MeetingType$outboundSchema` instead. */
    MeetingType$.outboundSchema = exports.MeetingType$outboundSchema;
})(MeetingType$ || (exports.MeetingType$ = MeetingType$ = {}));
/** @internal */
exports.ListMeetingsRequest$inboundSchema = z.object({
    "calendar_invitees[]": z.array(z.string()).optional(),
    "calendar_invitees_domains[]": z.array(z.string()).optional(),
    calendar_invitees_domains_type: exports.CalendarInviteesDomainsType$inboundSchema
        .default("all"),
    created_after: z.string().optional(),
    created_before: z.string().optional(),
    cursor: z.string().optional(),
    include_action_items: z.boolean().default(false),
    include_crm_matches: z.boolean().default(false),
    include_summary: z.boolean().default(false),
    include_transcript: z.boolean().default(false),
    meeting_type: exports.MeetingType$inboundSchema.optional(),
    "recorded_by[]": z.array(z.string()).optional(),
    "teams[]": z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "calendar_invitees[]": "calendarInvitees",
        "calendar_invitees_domains[]": "calendarInviteesDomains",
        "calendar_invitees_domains_type": "calendarInviteesDomainsType",
        "created_after": "createdAfter",
        "created_before": "createdBefore",
        "include_action_items": "includeActionItems",
        "include_crm_matches": "includeCrmMatches",
        "include_summary": "includeSummary",
        "include_transcript": "includeTranscript",
        "meeting_type": "meetingType",
        "recorded_by[]": "recordedBy",
        "teams[]": "teams",
    });
});
/** @internal */
exports.ListMeetingsRequest$outboundSchema = z.object({
    calendarInvitees: z.array(z.string()).optional(),
    calendarInviteesDomains: z.array(z.string()).optional(),
    calendarInviteesDomainsType: exports.CalendarInviteesDomainsType$outboundSchema
        .default("all"),
    createdAfter: z.string().optional(),
    createdBefore: z.string().optional(),
    cursor: z.string().optional(),
    includeActionItems: z.boolean().default(false),
    includeCrmMatches: z.boolean().default(false),
    includeSummary: z.boolean().default(false),
    includeTranscript: z.boolean().default(false),
    meetingType: exports.MeetingType$outboundSchema.optional(),
    recordedBy: z.array(z.string()).optional(),
    teams: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        calendarInvitees: "calendar_invitees[]",
        calendarInviteesDomains: "calendar_invitees_domains[]",
        calendarInviteesDomainsType: "calendar_invitees_domains_type",
        createdAfter: "created_after",
        createdBefore: "created_before",
        includeActionItems: "include_action_items",
        includeCrmMatches: "include_crm_matches",
        includeSummary: "include_summary",
        includeTranscript: "include_transcript",
        meetingType: "meeting_type",
        recordedBy: "recorded_by[]",
        teams: "teams[]",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListMeetingsRequest$;
(function (ListMeetingsRequest$) {
    /** @deprecated use `ListMeetingsRequest$inboundSchema` instead. */
    ListMeetingsRequest$.inboundSchema = exports.ListMeetingsRequest$inboundSchema;
    /** @deprecated use `ListMeetingsRequest$outboundSchema` instead. */
    ListMeetingsRequest$.outboundSchema = exports.ListMeetingsRequest$outboundSchema;
})(ListMeetingsRequest$ || (exports.ListMeetingsRequest$ = ListMeetingsRequest$ = {}));
function listMeetingsRequestToJSON(listMeetingsRequest) {
    return JSON.stringify(exports.ListMeetingsRequest$outboundSchema.parse(listMeetingsRequest));
}
function listMeetingsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListMeetingsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListMeetingsRequest' from JSON`);
}
/** @internal */
exports.ListMeetingsResponse$inboundSchema = z.object({
    Result: shared.MeetingListResponse$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ListMeetingsResponse$outboundSchema = z.object({
    result: shared.MeetingListResponse$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListMeetingsResponse$;
(function (ListMeetingsResponse$) {
    /** @deprecated use `ListMeetingsResponse$inboundSchema` instead. */
    ListMeetingsResponse$.inboundSchema = exports.ListMeetingsResponse$inboundSchema;
    /** @deprecated use `ListMeetingsResponse$outboundSchema` instead. */
    ListMeetingsResponse$.outboundSchema = exports.ListMeetingsResponse$outboundSchema;
})(ListMeetingsResponse$ || (exports.ListMeetingsResponse$ = ListMeetingsResponse$ = {}));
function listMeetingsResponseToJSON(listMeetingsResponse) {
    return JSON.stringify(exports.ListMeetingsResponse$outboundSchema.parse(listMeetingsResponse));
}
function listMeetingsResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListMeetingsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListMeetingsResponse' from JSON`);
}
