// fathom-typescript/dist/commonjs/sdk/models/operations/listteammembers.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTeamMembersResponse$ = exports.ListTeamMembersResponse$outboundSchema = exports.ListTeamMembersResponse$inboundSchema = exports.ListTeamMembersRequest$ = exports.ListTeamMembersRequest$outboundSchema = exports.ListTeamMembersRequest$inboundSchema = void 0;
exports.listTeamMembersRequestToJSON = listTeamMembersRequestToJSON;
exports.listTeamMembersRequestFromJSON = listTeamMembersRequestFromJSON;
exports.listTeamMembersResponseToJSON = listTeamMembersResponseToJSON;
exports.listTeamMembersResponseFromJSON = listTeamMembersResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../../lib/primitives.js");
const schemas_js_1 = require("../../../lib/schemas.js");
const shared = __importStar(require("../shared/index.js"));
/** @internal */
exports.ListTeamMembersRequest$inboundSchema = z.object({
    cursor: z.string().optional(),
    team: z.string().optional(),
});
/** @internal */
exports.ListTeamMembersRequest$outboundSchema = z.object({
    cursor: z.string().optional(),
    team: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListTeamMembersRequest$;
(function (ListTeamMembersRequest$) {
    /** @deprecated use `ListTeamMembersRequest$inboundSchema` instead. */
    ListTeamMembersRequest$.inboundSchema = exports.ListTeamMembersRequest$inboundSchema;
    /** @deprecated use `ListTeamMembersRequest$outboundSchema` instead. */
    ListTeamMembersRequest$.outboundSchema = exports.ListTeamMembersRequest$outboundSchema;
})(ListTeamMembersRequest$ || (exports.ListTeamMembersRequest$ = ListTeamMembersRequest$ = {}));
function listTeamMembersRequestToJSON(listTeamMembersRequest) {
    return JSON.stringify(exports.ListTeamMembersRequest$outboundSchema.parse(listTeamMembersRequest));
}
function listTeamMembersRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListTeamMembersRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamMembersRequest' from JSON`);
}
/** @internal */
exports.ListTeamMembersResponse$inboundSchema = z.object({
    Result: shared.TeamMemberListResponse$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.ListTeamMembersResponse$outboundSchema = z.object({
    result: shared.TeamMemberListResponse$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListTeamMembersResponse$;
(function (ListTeamMembersResponse$) {
    /** @deprecated use `ListTeamMembersResponse$inboundSchema` instead. */
    ListTeamMembersResponse$.inboundSchema = exports.ListTeamMembersResponse$inboundSchema;
    /** @deprecated use `ListTeamMembersResponse$outboundSchema` instead. */
    ListTeamMembersResponse$.outboundSchema = exports.ListTeamMembersResponse$outboundSchema;
})(ListTeamMembersResponse$ || (exports.ListTeamMembersResponse$ = ListTeamMembersResponse$ = {}));
function listTeamMembersResponseToJSON(listTeamMembersResponse) {
    return JSON.stringify(exports.ListTeamMembersResponse$outboundSchema.parse(listTeamMembersResponse));
}
function listTeamMembersResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListTeamMembersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListTeamMembersResponse' from JSON`);
}
