// fathom-typescript/src/sdk/models/operations/createwebhook.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { ClosedEnum } from "../../types/enums.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export const TriggeredFor = {
  MyRecordings: "my_recordings",
  SharedExternalRecordings: "shared_external_recordings",
  MySharedWithTeamRecordings: "my_shared_with_team_recordings",
  SharedTeamRecordings: "shared_team_recordings",
} as const;
export type TriggeredFor = ClosedEnum<typeof TriggeredFor>;

export type CreateWebhookRequest = {
  /**
   * The URL to send the webhook to.
   */
  destinationUrl: string;
  /**
   * Include the action items for each meeting.
   */
  includeActionItems?: boolean | undefined;
  /**
   * Include CRM matches for each meeting. Only returns data from your or your team's linked CRM.
   */
  includeCrmMatches?: boolean | undefined;
  /**
   * Include the summary for each meeting.
   */
  includeSummary?: boolean | undefined;
  /**
   * Include the transcript for each meeting.
   */
  includeTranscript?: boolean | undefined;
  /**
   * You must send at least one of the following types of recordings to trigger on.
   *
   * @remarks
   * - `my_recordings`: Your private recordings, as well as those you've shared with individuals. (On Team Plans, this excludes recordings you've shared with any teams.)
   * - `shared_external_recordings`: Recordings shared with you by other users. (For Team Plans, this does not include recordings shared by other users on your Team Plan.)
   * - `my_shared_with_team_recordings`: (Team Plans only). All recordings that you have shared with other teams (e.g. Marketing or Sales). Recordings you've shared with individuals but not with teams are not included.
   * - `shared_team_recordings`: (Team Plans only) All recordings you can access from other users on your Team Plan, whether shared with you individually or with your team.
   */
  triggeredFor: Array<TriggeredFor>;
};

/** @internal */
export const TriggeredFor$inboundSchema: z.ZodNativeEnum<typeof TriggeredFor> =
  z.nativeEnum(TriggeredFor);

/** @internal */
export const TriggeredFor$outboundSchema: z.ZodNativeEnum<typeof TriggeredFor> =
  TriggeredFor$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TriggeredFor$ {
  /** @deprecated use `TriggeredFor$inboundSchema` instead. */
  export const inboundSchema = TriggeredFor$inboundSchema;
  /** @deprecated use `TriggeredFor$outboundSchema` instead. */
  export const outboundSchema = TriggeredFor$outboundSchema;
}

/** @internal */
export const CreateWebhookRequest$inboundSchema: z.ZodType<
  CreateWebhookRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  destination_url: z.string(),
  include_action_items: z.boolean().default(false),
  include_crm_matches: z.boolean().default(false),
  include_summary: z.boolean().default(false),
  include_transcript: z.boolean().default(false),
  triggered_for: z.array(TriggeredFor$inboundSchema),
}).transform((v) => {
  return remap$(v, {
    "destination_url": "destinationUrl",
    "include_action_items": "includeActionItems",
    "include_crm_matches": "includeCrmMatches",
    "include_summary": "includeSummary",
    "include_transcript": "includeTranscript",
    "triggered_for": "triggeredFor",
  });
});

/** @internal */
export type CreateWebhookRequest$Outbound = {
  destination_url: string;
  include_action_items: boolean;
  include_crm_matches: boolean;
  include_summary: boolean;
  include_transcript: boolean;
  triggered_for: Array<string>;
};

/** @internal */
export const CreateWebhookRequest$outboundSchema: z.ZodType<
  CreateWebhookRequest$Outbound,
  z.ZodTypeDef,
  CreateWebhookRequest
> = z.object({
  destinationUrl: z.string(),
  includeActionItems: z.boolean().default(false),
  includeCrmMatches: z.boolean().default(false),
  includeSummary: z.boolean().default(false),
  includeTranscript: z.boolean().default(false),
  triggeredFor: z.array(TriggeredFor$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    destinationUrl: "destination_url",
    includeActionItems: "include_action_items",
    includeCrmMatches: "include_crm_matches",
    includeSummary: "include_summary",
    includeTranscript: "include_transcript",
    triggeredFor: "triggered_for",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CreateWebhookRequest$ {
  /** @deprecated use `CreateWebhookRequest$inboundSchema` instead. */
  export const inboundSchema = CreateWebhookRequest$inboundSchema;
  /** @deprecated use `CreateWebhookRequest$outboundSchema` instead. */
  export const outboundSchema = CreateWebhookRequest$outboundSchema;
  /** @deprecated use `CreateWebhookRequest$Outbound` instead. */
  export type Outbound = CreateWebhookRequest$Outbound;
}

export function createWebhookRequestToJSON(
  createWebhookRequest: CreateWebhookRequest,
): string {
  return JSON.stringify(
    CreateWebhookRequest$outboundSchema.parse(createWebhookRequest),
  );
}

export function createWebhookRequestFromJSON(
  jsonString: string,
): SafeParseResult<CreateWebhookRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CreateWebhookRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CreateWebhookRequest' from JSON`,
  );
}
