// fathom-typescript/src/sdk/models/operations/getrecordingsummary.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import * as shared from "../shared/index.js";

export type GetRecordingSummaryRequest = {
  /**
   * The ID of the meeting recording to fetch the call summary for.
   */
  recordingId: number;
  /**
   * Destination URL for where we'll POST the call summary. If not sent, this endpoint will return the data directly.
   */
  destinationUrl?: string | undefined;
};

export type GetRecordingSummaryResponseBody = {
  summary: shared.MeetingSummary | null;
};

/**
 * Either the destination URL for where we'll POST the call summary, or the summary for the recording.
 */
export type GetRecordingSummaryResponse =
  | GetRecordingSummaryResponseBody
  | shared.CallbackResponse;

/** @internal */
export const GetRecordingSummaryRequest$inboundSchema: z.ZodType<
  GetRecordingSummaryRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  recording_id: z.number().int(),
  destination_url: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "recording_id": "recordingId",
    "destination_url": "destinationUrl",
  });
});

/** @internal */
export type GetRecordingSummaryRequest$Outbound = {
  recording_id: number;
  destination_url?: string | undefined;
};

/** @internal */
export const GetRecordingSummaryRequest$outboundSchema: z.ZodType<
  GetRecordingSummaryRequest$Outbound,
  z.ZodTypeDef,
  GetRecordingSummaryRequest
> = z.object({
  recordingId: z.number().int(),
  destinationUrl: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    recordingId: "recording_id",
    destinationUrl: "destination_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetRecordingSummaryRequest$ {
  /** @deprecated use `GetRecordingSummaryRequest$inboundSchema` instead. */
  export const inboundSchema = GetRecordingSummaryRequest$inboundSchema;
  /** @deprecated use `GetRecordingSummaryRequest$outboundSchema` instead. */
  export const outboundSchema = GetRecordingSummaryRequest$outboundSchema;
  /** @deprecated use `GetRecordingSummaryRequest$Outbound` instead. */
  export type Outbound = GetRecordingSummaryRequest$Outbound;
}

export function getRecordingSummaryRequestToJSON(
  getRecordingSummaryRequest: GetRecordingSummaryRequest,
): string {
  return JSON.stringify(
    GetRecordingSummaryRequest$outboundSchema.parse(getRecordingSummaryRequest),
  );
}

export function getRecordingSummaryRequestFromJSON(
  jsonString: string,
): SafeParseResult<GetRecordingSummaryRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => GetRecordingSummaryRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetRecordingSummaryRequest' from JSON`,
  );
}

/** @internal */
export const GetRecordingSummaryResponseBody$inboundSchema: z.ZodType<
  GetRecordingSummaryResponseBody,
  z.ZodTypeDef,
  unknown
> = z.object({
  summary: z.nullable(shared.MeetingSummary$inboundSchema),
});

/** @internal */
export type GetRecordingSummaryResponseBody$Outbound = {
  summary: shared.MeetingSummary$Outbound | null;
};

/** @internal */
export const GetRecordingSummaryResponseBody$outboundSchema: z.ZodType<
  GetRecordingSummaryResponseBody$Outbound,
  z.ZodTypeDef,
  GetRecordingSummaryResponseBody
> = z.object({
  summary: z.nullable(shared.MeetingSummary$outboundSchema),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetRecordingSummaryResponseBody$ {
  /** @deprecated use `GetRecordingSummaryResponseBody$inboundSchema` instead. */
  export const inboundSchema = GetRecordingSummaryResponseBody$inboundSchema;
  /** @deprecated use `GetRecordingSummaryResponseBody$outboundSchema` instead. */
  export const outboundSchema = GetRecordingSummaryResponseBody$outboundSchema;
  /** @deprecated use `GetRecordingSummaryResponseBody$Outbound` instead. */
  export type Outbound = GetRecordingSummaryResponseBody$Outbound;
}

export function getRecordingSummaryResponseBodyToJSON(
  getRecordingSummaryResponseBody: GetRecordingSummaryResponseBody,
): string {
  return JSON.stringify(
    GetRecordingSummaryResponseBody$outboundSchema.parse(
      getRecordingSummaryResponseBody,
    ),
  );
}

export function getRecordingSummaryResponseBodyFromJSON(
  jsonString: string,
): SafeParseResult<GetRecordingSummaryResponseBody, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => GetRecordingSummaryResponseBody$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetRecordingSummaryResponseBody' from JSON`,
  );
}

/** @internal */
export const GetRecordingSummaryResponse$inboundSchema: z.ZodType<
  GetRecordingSummaryResponse,
  z.ZodTypeDef,
  unknown
> = z.union([
  z.lazy(() => GetRecordingSummaryResponseBody$inboundSchema),
  shared.CallbackResponse$inboundSchema,
]);

/** @internal */
export type GetRecordingSummaryResponse$Outbound =
  | GetRecordingSummaryResponseBody$Outbound
  | shared.CallbackResponse$Outbound;

/** @internal */
export const GetRecordingSummaryResponse$outboundSchema: z.ZodType<
  GetRecordingSummaryResponse$Outbound,
  z.ZodTypeDef,
  GetRecordingSummaryResponse
> = z.union([
  z.lazy(() => GetRecordingSummaryResponseBody$outboundSchema),
  shared.CallbackResponse$outboundSchema,
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetRecordingSummaryResponse$ {
  /** @deprecated use `GetRecordingSummaryResponse$inboundSchema` instead. */
  export const inboundSchema = GetRecordingSummaryResponse$inboundSchema;
  /** @deprecated use `GetRecordingSummaryResponse$outboundSchema` instead. */
  export const outboundSchema = GetRecordingSummaryResponse$outboundSchema;
  /** @deprecated use `GetRecordingSummaryResponse$Outbound` instead. */
  export type Outbound = GetRecordingSummaryResponse$Outbound;
}

export function getRecordingSummaryResponseToJSON(
  getRecordingSummaryResponse: GetRecordingSummaryResponse,
): string {
  return JSON.stringify(
    GetRecordingSummaryResponse$outboundSchema.parse(
      getRecordingSummaryResponse,
    ),
  );
}

export function getRecordingSummaryResponseFromJSON(
  jsonString: string,
): SafeParseResult<GetRecordingSummaryResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => GetRecordingSummaryResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetRecordingSummaryResponse' from JSON`,
  );
}
