// fathom-typescript/src/sdk/models/operations/getrecordingtranscript.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import * as shared from "../shared/index.js";

export type GetRecordingTranscriptRequest = {
  /**
   * The ID of the meeting recording to fetch the transcript for.
   */
  recordingId: number;
  /**
   * Destination URL for where we'll POST the transcript. If not sent, this endpoint will return the data directly.
   */
  destinationUrl: string;
};

export type GetRecordingTranscriptResponseBody = {
  transcript: Array<shared.TranscriptItem>;
};

/**
 * Either the destination URL for where we'll POST the transcript, or the transcript for the recording as an array.
 */
export type GetRecordingTranscriptResponse =
  | GetRecordingTranscriptResponseBody
  | shared.CallbackResponse;

/** @internal */
export const GetRecordingTranscriptRequest$inboundSchema: z.ZodType<
  GetRecordingTranscriptRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  recording_id: z.number().int(),
  destination_url: z.string(),
}).transform((v) => {
  return remap$(v, {
    "recording_id": "recordingId",
    "destination_url": "destinationUrl",
  });
});

/** @internal */
export type GetRecordingTranscriptRequest$Outbound = {
  recording_id: number;
  destination_url: string;
};

/** @internal */
export const GetRecordingTranscriptRequest$outboundSchema: z.ZodType<
  GetRecordingTranscriptRequest$Outbound,
  z.ZodTypeDef,
  GetRecordingTranscriptRequest
> = z.object({
  recordingId: z.number().int(),
  destinationUrl: z.string(),
}).transform((v) => {
  return remap$(v, {
    recordingId: "recording_id",
    destinationUrl: "destination_url",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetRecordingTranscriptRequest$ {
  /** @deprecated use `GetRecordingTranscriptRequest$inboundSchema` instead. */
  export const inboundSchema = GetRecordingTranscriptRequest$inboundSchema;
  /** @deprecated use `GetRecordingTranscriptRequest$outboundSchema` instead. */
  export const outboundSchema = GetRecordingTranscriptRequest$outboundSchema;
  /** @deprecated use `GetRecordingTranscriptRequest$Outbound` instead. */
  export type Outbound = GetRecordingTranscriptRequest$Outbound;
}

export function getRecordingTranscriptRequestToJSON(
  getRecordingTranscriptRequest: GetRecordingTranscriptRequest,
): string {
  return JSON.stringify(
    GetRecordingTranscriptRequest$outboundSchema.parse(
      getRecordingTranscriptRequest,
    ),
  );
}

export function getRecordingTranscriptRequestFromJSON(
  jsonString: string,
): SafeParseResult<GetRecordingTranscriptRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => GetRecordingTranscriptRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetRecordingTranscriptRequest' from JSON`,
  );
}

/** @internal */
export const GetRecordingTranscriptResponseBody$inboundSchema: z.ZodType<
  GetRecordingTranscriptResponseBody,
  z.ZodTypeDef,
  unknown
> = z.object({
  transcript: z.array(shared.TranscriptItem$inboundSchema),
});

/** @internal */
export type GetRecordingTranscriptResponseBody$Outbound = {
  transcript: Array<shared.TranscriptItem$Outbound>;
};

/** @internal */
export const GetRecordingTranscriptResponseBody$outboundSchema: z.ZodType<
  GetRecordingTranscriptResponseBody$Outbound,
  z.ZodTypeDef,
  GetRecordingTranscriptResponseBody
> = z.object({
  transcript: z.array(shared.TranscriptItem$outboundSchema),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetRecordingTranscriptResponseBody$ {
  /** @deprecated use `GetRecordingTranscriptResponseBody$inboundSchema` instead. */
  export const inboundSchema = GetRecordingTranscriptResponseBody$inboundSchema;
  /** @deprecated use `GetRecordingTranscriptResponseBody$outboundSchema` instead. */
  export const outboundSchema =
    GetRecordingTranscriptResponseBody$outboundSchema;
  /** @deprecated use `GetRecordingTranscriptResponseBody$Outbound` instead. */
  export type Outbound = GetRecordingTranscriptResponseBody$Outbound;
}

export function getRecordingTranscriptResponseBodyToJSON(
  getRecordingTranscriptResponseBody: GetRecordingTranscriptResponseBody,
): string {
  return JSON.stringify(
    GetRecordingTranscriptResponseBody$outboundSchema.parse(
      getRecordingTranscriptResponseBody,
    ),
  );
}

export function getRecordingTranscriptResponseBodyFromJSON(
  jsonString: string,
): SafeParseResult<GetRecordingTranscriptResponseBody, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) =>
      GetRecordingTranscriptResponseBody$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetRecordingTranscriptResponseBody' from JSON`,
  );
}

/** @internal */
export const GetRecordingTranscriptResponse$inboundSchema: z.ZodType<
  GetRecordingTranscriptResponse,
  z.ZodTypeDef,
  unknown
> = z.union([
  z.lazy(() => GetRecordingTranscriptResponseBody$inboundSchema),
  shared.CallbackResponse$inboundSchema,
]);

/** @internal */
export type GetRecordingTranscriptResponse$Outbound =
  | GetRecordingTranscriptResponseBody$Outbound
  | shared.CallbackResponse$Outbound;

/** @internal */
export const GetRecordingTranscriptResponse$outboundSchema: z.ZodType<
  GetRecordingTranscriptResponse$Outbound,
  z.ZodTypeDef,
  GetRecordingTranscriptResponse
> = z.union([
  z.lazy(() => GetRecordingTranscriptResponseBody$outboundSchema),
  shared.CallbackResponse$outboundSchema,
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace GetRecordingTranscriptResponse$ {
  /** @deprecated use `GetRecordingTranscriptResponse$inboundSchema` instead. */
  export const inboundSchema = GetRecordingTranscriptResponse$inboundSchema;
  /** @deprecated use `GetRecordingTranscriptResponse$outboundSchema` instead. */
  export const outboundSchema = GetRecordingTranscriptResponse$outboundSchema;
  /** @deprecated use `GetRecordingTranscriptResponse$Outbound` instead. */
  export type Outbound = GetRecordingTranscriptResponse$Outbound;
}

export function getRecordingTranscriptResponseToJSON(
  getRecordingTranscriptResponse: GetRecordingTranscriptResponse,
): string {
  return JSON.stringify(
    GetRecordingTranscriptResponse$outboundSchema.parse(
      getRecordingTranscriptResponse,
    ),
  );
}

export function getRecordingTranscriptResponseFromJSON(
  jsonString: string,
): SafeParseResult<GetRecordingTranscriptResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => GetRecordingTranscriptResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetRecordingTranscriptResponse' from JSON`,
  );
}
