// fathom-typescript/src/sdk/models/operations/listteammembers.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import * as shared from "../shared/index.js";

export type ListTeamMembersRequest = {
  /**
   * Cursor for pagination.
   */
  cursor?: string | undefined;
  /**
   * Team name to filter by.
   */
  team?: string | undefined;
};

export type ListTeamMembersResponse = {
  result: shared.TeamMemberListResponse;
};

/** @internal */
export const ListTeamMembersRequest$inboundSchema: z.ZodType<
  ListTeamMembersRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  cursor: z.string().optional(),
  team: z.string().optional(),
});

/** @internal */
export type ListTeamMembersRequest$Outbound = {
  cursor?: string | undefined;
  team?: string | undefined;
};

/** @internal */
export const ListTeamMembersRequest$outboundSchema: z.ZodType<
  ListTeamMembersRequest$Outbound,
  z.ZodTypeDef,
  ListTeamMembersRequest
> = z.object({
  cursor: z.string().optional(),
  team: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListTeamMembersRequest$ {
  /** @deprecated use `ListTeamMembersRequest$inboundSchema` instead. */
  export const inboundSchema = ListTeamMembersRequest$inboundSchema;
  /** @deprecated use `ListTeamMembersRequest$outboundSchema` instead. */
  export const outboundSchema = ListTeamMembersRequest$outboundSchema;
  /** @deprecated use `ListTeamMembersRequest$Outbound` instead. */
  export type Outbound = ListTeamMembersRequest$Outbound;
}

export function listTeamMembersRequestToJSON(
  listTeamMembersRequest: ListTeamMembersRequest,
): string {
  return JSON.stringify(
    ListTeamMembersRequest$outboundSchema.parse(listTeamMembersRequest),
  );
}

export function listTeamMembersRequestFromJSON(
  jsonString: string,
): SafeParseResult<ListTeamMembersRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListTeamMembersRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListTeamMembersRequest' from JSON`,
  );
}

/** @internal */
export const ListTeamMembersResponse$inboundSchema: z.ZodType<
  ListTeamMembersResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  Result: shared.TeamMemberListResponse$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "Result": "result",
  });
});

/** @internal */
export type ListTeamMembersResponse$Outbound = {
  Result: shared.TeamMemberListResponse$Outbound;
};

/** @internal */
export const ListTeamMembersResponse$outboundSchema: z.ZodType<
  ListTeamMembersResponse$Outbound,
  z.ZodTypeDef,
  ListTeamMembersResponse
> = z.object({
  result: shared.TeamMemberListResponse$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    result: "Result",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListTeamMembersResponse$ {
  /** @deprecated use `ListTeamMembersResponse$inboundSchema` instead. */
  export const inboundSchema = ListTeamMembersResponse$inboundSchema;
  /** @deprecated use `ListTeamMembersResponse$outboundSchema` instead. */
  export const outboundSchema = ListTeamMembersResponse$outboundSchema;
  /** @deprecated use `ListTeamMembersResponse$Outbound` instead. */
  export type Outbound = ListTeamMembersResponse$Outbound;
}

export function listTeamMembersResponseToJSON(
  listTeamMembersResponse: ListTeamMembersResponse,
): string {
  return JSON.stringify(
    ListTeamMembersResponse$outboundSchema.parse(listTeamMembersResponse),
  );
}

export function listTeamMembersResponseFromJSON(
  jsonString: string,
): SafeParseResult<ListTeamMembersResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListTeamMembersResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListTeamMembersResponse' from JSON`,
  );
}
