// fathom-typescript/src/sdk/models/operations/listmeetings.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { ClosedEnum } from "../../types/enums.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import * as shared from "../shared/index.js";

/**
 * Filter by whether calendar invitee list includes external email domains.
 */
export const CalendarInviteesDomainsType = {
  All: "all",
  OnlyInternal: "only_internal",
  OneOrMoreExternal: "one_or_more_external",
} as const;
/**
 * Filter by whether calendar invitee list includes external email domains.
 */
export type CalendarInviteesDomainsType = ClosedEnum<
  typeof CalendarInviteesDomainsType
>;

/**
 * Filter by meeting type.
 */
export const MeetingType = {
  All: "all",
  Internal: "internal",
  External: "external",
} as const;
/**
 * Filter by meeting type.
 */
export type MeetingType = ClosedEnum<typeof MeetingType>;

export type ListMeetingsRequest = {
  /**
   * Email address of calendar_invitees to filter by.
   *
   * @remarks
   *
   * Pass the parameter once per value, e.g.
   * `calendar_invitees[]=cfo@acme.com&calendar_invitees[]=legal@acme.com`.
   *
   * Returns meetings where any of the given email addresses appear
   * in the calendar_invitees list.
   */
  calendarInvitees?: Array<string> | undefined;
  /**
   * Domains of the companies to filter by. Exact match.
   *
   * @remarks
   *
   * Pass the parameter once per value, e.g.
   * `calendar_invitees_domains[]=acme.com&calendar_invitees_domains[]=client.com`.
   *
   * Returns meetings where any of the given company domains appear in the meeting.
   */
  calendarInviteesDomains?: Array<string> | undefined;
  /**
   * Filter by whether calendar invitee list includes external email domains.
   */
  calendarInviteesDomainsType?: CalendarInviteesDomainsType | undefined;
  /**
   * Filter to meetings with created_at after this timestamp, e.g. `created_after=2025-01-01T00:00:00Z`.
   */
  createdAfter?: string | undefined;
  /**
   * Filter to meetings with created_at before this timestamp, e.g. `created_before=2025-01-01T00:00:00Z`.
   */
  createdBefore?: string | undefined;
  /**
   * Cursor for pagination.
   */
  cursor?: string | undefined;
  /**
   * Include the action items for each meeting.
   */
  includeActionItems?: boolean | undefined;
  /**
   * Include CRM matches for each meeting. Only returns data from your or your team's linked CRM.
   */
  includeCrmMatches?: boolean | undefined;
  /**
   * Include the summary for each meeting. Unavailable for OAuth connected apps (use /recordings instead).
   */
  includeSummary?: boolean | undefined;
  /**
   * Include the transcript for each meeting. Unavailable for OAuth connected apps (use /recordings instead).
   */
  includeTranscript?: boolean | undefined;
  /**
   * Filter by meeting type.
   *
   * @deprecated field: This will be removed in a future release, please migrate away from it as soon as possible.
   */
  meetingType?: MeetingType | undefined;
  /**
   * Email addresses of users who recorded meetings.
   *
   * @remarks
   *
   * Pass the parameter once per value, e.g.
   * `recorded_by[]=ceo@acme.com&recorded_by[]=pm@acme.com`.
   *
   * Returns meetings recorded by any of the specified users.
   */
  recordedBy?: Array<string> | undefined;
  /**
   * Team names to filter by.
   *
   * @remarks
   *
   * Pass the parameter once per value, e.g.
   * `teams[]=Sales&teams[]=Engineering`.
   *
   * Returns meetings that belong to any of the specified teams.
   */
  teams?: Array<string> | undefined;
};

export type ListMeetingsResponse = {
  result: shared.MeetingListResponse;
};

/** @internal */
export const CalendarInviteesDomainsType$inboundSchema: z.ZodNativeEnum<
  typeof CalendarInviteesDomainsType
> = z.nativeEnum(CalendarInviteesDomainsType);

/** @internal */
export const CalendarInviteesDomainsType$outboundSchema: z.ZodNativeEnum<
  typeof CalendarInviteesDomainsType
> = CalendarInviteesDomainsType$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CalendarInviteesDomainsType$ {
  /** @deprecated use `CalendarInviteesDomainsType$inboundSchema` instead. */
  export const inboundSchema = CalendarInviteesDomainsType$inboundSchema;
  /** @deprecated use `CalendarInviteesDomainsType$outboundSchema` instead. */
  export const outboundSchema = CalendarInviteesDomainsType$outboundSchema;
}

/** @internal */
export const MeetingType$inboundSchema: z.ZodNativeEnum<typeof MeetingType> = z
  .nativeEnum(MeetingType);

/** @internal */
export const MeetingType$outboundSchema: z.ZodNativeEnum<typeof MeetingType> =
  MeetingType$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace MeetingType$ {
  /** @deprecated use `MeetingType$inboundSchema` instead. */
  export const inboundSchema = MeetingType$inboundSchema;
  /** @deprecated use `MeetingType$outboundSchema` instead. */
  export const outboundSchema = MeetingType$outboundSchema;
}

/** @internal */
export const ListMeetingsRequest$inboundSchema: z.ZodType<
  ListMeetingsRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  "calendar_invitees[]": z.array(z.string()).optional(),
  "calendar_invitees_domains[]": z.array(z.string()).optional(),
  calendar_invitees_domains_type: CalendarInviteesDomainsType$inboundSchema
    .default("all"),
  created_after: z.string().optional(),
  created_before: z.string().optional(),
  cursor: z.string().optional(),
  include_action_items: z.boolean().default(false),
  include_crm_matches: z.boolean().default(false),
  include_summary: z.boolean().default(false),
  include_transcript: z.boolean().default(false),
  meeting_type: MeetingType$inboundSchema.optional(),
  "recorded_by[]": z.array(z.string()).optional(),
  "teams[]": z.array(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    "calendar_invitees[]": "calendarInvitees",
    "calendar_invitees_domains[]": "calendarInviteesDomains",
    "calendar_invitees_domains_type": "calendarInviteesDomainsType",
    "created_after": "createdAfter",
    "created_before": "createdBefore",
    "include_action_items": "includeActionItems",
    "include_crm_matches": "includeCrmMatches",
    "include_summary": "includeSummary",
    "include_transcript": "includeTranscript",
    "meeting_type": "meetingType",
    "recorded_by[]": "recordedBy",
    "teams[]": "teams",
  });
});

/** @internal */
export type ListMeetingsRequest$Outbound = {
  "calendar_invitees[]"?: Array<string> | undefined;
  "calendar_invitees_domains[]"?: Array<string> | undefined;
  calendar_invitees_domains_type: string;
  created_after?: string | undefined;
  created_before?: string | undefined;
  cursor?: string | undefined;
  include_action_items: boolean;
  include_crm_matches: boolean;
  include_summary: boolean;
  include_transcript: boolean;
  meeting_type?: string | undefined;
  "recorded_by[]"?: Array<string> | undefined;
  "teams[]"?: Array<string> | undefined;
};

/** @internal */
export const ListMeetingsRequest$outboundSchema: z.ZodType<
  ListMeetingsRequest$Outbound,
  z.ZodTypeDef,
  ListMeetingsRequest
> = z.object({
  calendarInvitees: z.array(z.string()).optional(),
  calendarInviteesDomains: z.array(z.string()).optional(),
  calendarInviteesDomainsType: CalendarInviteesDomainsType$outboundSchema
    .default("all"),
  createdAfter: z.string().optional(),
  createdBefore: z.string().optional(),
  cursor: z.string().optional(),
  includeActionItems: z.boolean().default(false),
  includeCrmMatches: z.boolean().default(false),
  includeSummary: z.boolean().default(false),
  includeTranscript: z.boolean().default(false),
  meetingType: MeetingType$outboundSchema.optional(),
  recordedBy: z.array(z.string()).optional(),
  teams: z.array(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    calendarInvitees: "calendar_invitees[]",
    calendarInviteesDomains: "calendar_invitees_domains[]",
    calendarInviteesDomainsType: "calendar_invitees_domains_type",
    createdAfter: "created_after",
    createdBefore: "created_before",
    includeActionItems: "include_action_items",
    includeCrmMatches: "include_crm_matches",
    includeSummary: "include_summary",
    includeTranscript: "include_transcript",
    meetingType: "meeting_type",
    recordedBy: "recorded_by[]",
    teams: "teams[]",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListMeetingsRequest$ {
  /** @deprecated use `ListMeetingsRequest$inboundSchema` instead. */
  export const inboundSchema = ListMeetingsRequest$inboundSchema;
  /** @deprecated use `ListMeetingsRequest$outboundSchema` instead. */
  export const outboundSchema = ListMeetingsRequest$outboundSchema;
  /** @deprecated use `ListMeetingsRequest$Outbound` instead. */
  export type Outbound = ListMeetingsRequest$Outbound;
}

export function listMeetingsRequestToJSON(
  listMeetingsRequest: ListMeetingsRequest,
): string {
  return JSON.stringify(
    ListMeetingsRequest$outboundSchema.parse(listMeetingsRequest),
  );
}

export function listMeetingsRequestFromJSON(
  jsonString: string,
): SafeParseResult<ListMeetingsRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListMeetingsRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListMeetingsRequest' from JSON`,
  );
}

/** @internal */
export const ListMeetingsResponse$inboundSchema: z.ZodType<
  ListMeetingsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  Result: shared.MeetingListResponse$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "Result": "result",
  });
});

/** @internal */
export type ListMeetingsResponse$Outbound = {
  Result: shared.MeetingListResponse$Outbound;
};

/** @internal */
export const ListMeetingsResponse$outboundSchema: z.ZodType<
  ListMeetingsResponse$Outbound,
  z.ZodTypeDef,
  ListMeetingsResponse
> = z.object({
  result: shared.MeetingListResponse$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    result: "Result",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListMeetingsResponse$ {
  /** @deprecated use `ListMeetingsResponse$inboundSchema` instead. */
  export const inboundSchema = ListMeetingsResponse$inboundSchema;
  /** @deprecated use `ListMeetingsResponse$outboundSchema` instead. */
  export const outboundSchema = ListMeetingsResponse$outboundSchema;
  /** @deprecated use `ListMeetingsResponse$Outbound` instead. */
  export type Outbound = ListMeetingsResponse$Outbound;
}

export function listMeetingsResponseToJSON(
  listMeetingsResponse: ListMeetingsResponse,
): string {
  return JSON.stringify(
    ListMeetingsResponse$outboundSchema.parse(listMeetingsResponse),
  );
}

export function listMeetingsResponseFromJSON(
  jsonString: string,
): SafeParseResult<ListMeetingsResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListMeetingsResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListMeetingsResponse' from JSON`,
  );
}
