// fathom-typescript/src/sdk/models/operations/listteams.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../../lib/primitives.js";
import { safeParse } from "../../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import * as shared from "../shared/index.js";

export type ListTeamsRequest = {
  /**
   * Cursor for pagination.
   */
  cursor?: string | undefined;
};

export type ListTeamsResponse = {
  result: shared.TeamListResponse;
};

/** @internal */
export const ListTeamsRequest$inboundSchema: z.ZodType<
  ListTeamsRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  cursor: z.string().optional(),
});

/** @internal */
export type ListTeamsRequest$Outbound = {
  cursor?: string | undefined;
};

/** @internal */
export const ListTeamsRequest$outboundSchema: z.ZodType<
  ListTeamsRequest$Outbound,
  z.ZodTypeDef,
  ListTeamsRequest
> = z.object({
  cursor: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListTeamsRequest$ {
  /** @deprecated use `ListTeamsRequest$inboundSchema` instead. */
  export const inboundSchema = ListTeamsRequest$inboundSchema;
  /** @deprecated use `ListTeamsRequest$outboundSchema` instead. */
  export const outboundSchema = ListTeamsRequest$outboundSchema;
  /** @deprecated use `ListTeamsRequest$Outbound` instead. */
  export type Outbound = ListTeamsRequest$Outbound;
}

export function listTeamsRequestToJSON(
  listTeamsRequest: ListTeamsRequest,
): string {
  return JSON.stringify(
    ListTeamsRequest$outboundSchema.parse(listTeamsRequest),
  );
}

export function listTeamsRequestFromJSON(
  jsonString: string,
): SafeParseResult<ListTeamsRequest, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListTeamsRequest$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListTeamsRequest' from JSON`,
  );
}

/** @internal */
export const ListTeamsResponse$inboundSchema: z.ZodType<
  ListTeamsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  Result: shared.TeamListResponse$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "Result": "result",
  });
});

/** @internal */
export type ListTeamsResponse$Outbound = {
  Result: shared.TeamListResponse$Outbound;
};

/** @internal */
export const ListTeamsResponse$outboundSchema: z.ZodType<
  ListTeamsResponse$Outbound,
  z.ZodTypeDef,
  ListTeamsResponse
> = z.object({
  result: shared.TeamListResponse$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    result: "Result",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListTeamsResponse$ {
  /** @deprecated use `ListTeamsResponse$inboundSchema` instead. */
  export const inboundSchema = ListTeamsResponse$inboundSchema;
  /** @deprecated use `ListTeamsResponse$outboundSchema` instead. */
  export const outboundSchema = ListTeamsResponse$outboundSchema;
  /** @deprecated use `ListTeamsResponse$Outbound` instead. */
  export type Outbound = ListTeamsResponse$Outbound;
}

export function listTeamsResponseToJSON(
  listTeamsResponse: ListTeamsResponse,
): string {
  return JSON.stringify(
    ListTeamsResponse$outboundSchema.parse(listTeamsResponse),
  );
}

export function listTeamsResponseFromJSON(
  jsonString: string,
): SafeParseResult<ListTeamsResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListTeamsResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListTeamsResponse' from JSON`,
  );
}
